
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.6.3">
    
    
      
        <title>Variant Visualisation - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-56ade86843.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-e2807e330f.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-f78e5cb881.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/spoilers.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Cancer Genomics Modules
                </span>
              
            
            Variant Visualisation
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link md-tabs__link--active">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Variant Visualisation
      </label>
    
    <a href="./" title="Variant Visualisation" class="md-nav__link md-nav__link--active">
      Variant Visualisation
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" title="Acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#license" title="License" class="md-nav__link">
    License
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#acknowledgements" title="Acknowledgements" class="md-nav__link">
    Acknowledgements
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#license" title="License" class="md-nav__link">
    License
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Variant Visualisation</h1>
                
                <p>This module was written by Mathieu Bourgey and the original on-line version is available
<a href="https://github.com/mbourgey/EBI_cancer_workshop_visualization/tree/australia_2015">here.</a></p>
<hr />
<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>Generate Circos like graphics using R</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>R:<br />
<a href="https://cran.r-project.org/">https://cran.r-project.org/</a></p>
<p>R package circlize:<br />
<a href="https://cran.r-project.org/web/packages/circlize/index.html">https://cran.r-project.org/web/packages/circlize/index.html</a></p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>This short workshop will show you how to visualize your data.</p>
<p>We will be working on 3 types of somatic calls:</p>
<ul>
<li>
<p>SNV calls from MuTect (vcf)</p>
</li>
<li>
<p>SV calls from DELLY (vcf)</p>
</li>
<li>
<p>CNV calls from SCoNEs (tsv)</p>
</li>
</ul>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>We will use a dataset derived from the analysis of whole genome sequencing
paired normal/tumour samples.</p>
<p>The call files are contained in the folder <code class="codehilite">visualization</code>:</p>
<ul>
<li><code class="codehilite">mutect.somatic.vcf</code>  </li>
<li><code class="codehilite">delly.somatic.vcf</code>    </li>
<li><code class="codehilite">scones.somatic.tsv</code>   </li>
</ul>
<p>Many tools are available to do this and the most commonly known is Circos.
Circos is a really not user-friendly. In this tutorial, we show you an
easy alternative to build a circular representation of genomic data.</p>
<p>First we need to go in the folder <code class="codehilite">visualization</code> to do the analysis:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd /home/trainee/visualization/
</pre></div>
</td></tr></table>

<p>Let&rsquo;s see what is in this folder:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ls
</pre></div>
</td></tr></table>

<p><code class="codehilite">circos.R  delly.somatic.vcf  mutect.somatic.vcf  scones.somatic.tsv</code></p>
<p>Take a look at the data files.</p>
<p>This data has <em>not</em> been restricted to a short piece of a chromosome and SNVs
have already been filtered:</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less mutect.somatic.tsv
less delly.somatic.vcf
less data/scones.somatic.30k.tsv
</pre></div>
</td></tr></table>
<br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What can you see from this data?</p>
<div class="admonition success"><details class="spoilers"><summary><strong>Answer</strong></summary><ul><li><p>the filtered output of 3 different software: Mutect (SNVs),
Delly (SVs), SCoNEs (CNVs).</p></li><li><p>The 3 files show 2 different formats (vcf, tsv).</p></li><li><p>Almost all type of variants are represented here: mutations,
deletion, inversion, translocation, large amplification and deletion
(CNVs).</p></li></ul></details></div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why don’t we use the vcf format for all types of calls?</p>
<div class="admonition success"><details class="spoilers"><summary><strong>Answer</strong></summary><p>The 1000 Genomes project tries to use/include SV calls in the vcf format.
Some tools like Delly use this format for SVs. It is a good idea to try to
include everything together but this not the be the best way to handle SVs and
CNVs.</p></details></div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why?</p>
<div class="admonition success"><details class="spoilers"><summary><strong>Answer</strong></summary><p>Due to the nature of these calls, you can not easily integrate the
positional information of the two breakpoints (that could be located
far away or in an other chromosome) using a single position format.</p></details></div>
</div>
<p><br></p>
<p>The analysis will be done using the R program:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>R
</pre></div>
</td></tr></table>

<p>We will use the circlize package from the cran R project. This package generates
circular plots and has the advantage of being able to provide
pre-built functions for genomic data. One of the main advantages of this
tool is the use of bed format as input data.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>library(circlize)
</pre></div>
</td></tr></table>

<p><br>
Let’s import the variants:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>snp<span class="o">=</span>read.table<span class="p">(</span><span class="s">&quot;mutect.somatic.vcf&quot;</span><span class="p">)</span>
sv<span class="o">=</span>read.table<span class="p">(</span><span class="s">&quot;somatic.sv.vcf&quot;</span><span class="p">)</span>
cnv<span class="o">=</span>read.table<span class="p">(</span><span class="s">&quot;data/scones.somatic.tsv&quot;</span><span class="p">,</span>header<span class="o">=</span><span class="bp">T</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p><br>
We need to set up the generic graphical parameters:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>par<span class="p">(</span>mar <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">1</span><span class="p">,</span> <span class="m">1</span><span class="p">))</span>
circos.par<span class="p">(</span><span class="s">&quot;start.degree&quot;</span> <span class="o">=</span> <span class="m">90</span><span class="p">)</span>
circos.par<span class="p">(</span><span class="s">&quot;track.height&quot;</span> <span class="o">=</span> <span class="m">0.05</span><span class="p">)</span>
circos.par<span class="p">(</span><span class="s">&quot;canvas.xlim&quot;</span> <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">-1.3</span><span class="p">,</span> <span class="m">1.3</span><span class="p">),</span> <span class="s">&quot;canvas.ylim&quot;</span> <span class="o">=</span> <span class="kt">c</span><span class="p">(</span><span class="m">-1.3</span><span class="p">,</span> <span class="m">1.3</span><span class="p">))</span>
</pre></div>
</td></tr></table>

<p><br>
Let’s draw hg19 reference ideograms:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>circos.initializeWithIdeogram<span class="p">(</span>species <span class="o">=</span> <span class="s">&quot;hg19&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p><br>
Unfortunately circlize does not support hg38 yet. So we will need to
reformat our data to fit the hg19 standards. As we are working only on autosomes,
we won’t need to lift-over and we could simply add <em>chr</em> at the beginning
of the chromosome names.</p>
<p>We can now draw 1 track for somatic mutations:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>snv_tmp<span class="o">=</span>read.table<span class="p">(</span><span class="s">&quot;data/mutec.somatic.vcf&quot;</span><span class="p">,</span>comment.char<span class="o">=</span><span class="s">&quot;#&quot;</span><span class="p">)</span>
snv<span class="o">=</span><span class="kp">cbind</span><span class="p">(</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="kp">as.character</span><span class="p">(</span>snp<span class="p">[,</span><span class="m">1</span><span class="p">]),</span>sep<span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span>snp<span class="p">[</span><span class="m">2</span><span class="p">],</span>snp<span class="p">[,</span><span class="m">2</span><span class="p">]</span><span class="m">+1</span><span class="p">)</span>
circos.genomicTrackPlotRegion<span class="p">(</span>snv<span class="p">,</span>stack<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> panel.fun <span class="o">=</span> <span class="kr">function</span><span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> <span class="kc">...</span><span class="p">)</span> <span class="p">{</span>
    circos.genomicPoints<span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> cex <span class="o">=</span> <span class="m">0.05</span><span class="p">,</span> pch <span class="o">=</span> <span class="m">9</span><span class="p">,</span>col<span class="o">=</span><span class="s">&#39;orange&#39;</span> <span class="p">,</span> <span class="kc">...</span><span class="p">)</span>
<span class="p">})</span>
</pre></div>
</td></tr></table>

<p><br>
Let’s draw the 2 tracks for CNVs. One track for duplications in red and
one blue track for deletions.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>dup<span class="o">=</span>cnv<span class="p">[</span>cnv<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">&gt;</span><span class="m">2</span><span class="p">,]</span>
dup<span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="o">=</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="kp">as.character</span><span class="p">(</span>dup<span class="p">[,</span><span class="m">1</span><span class="p">]),</span>sep<span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>
del<span class="o">=</span>cnv<span class="p">[</span>cnv<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">&lt;</span><span class="m">2</span><span class="p">,]</span>
del<span class="p">[,</span><span class="m">1</span><span class="p">]</span><span class="o">=</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span><span class="kp">as.character</span><span class="p">(</span>del<span class="p">[,</span><span class="m">1</span><span class="p">]),</span>sep<span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">)</span>
circos.genomicTrackPlotRegion<span class="p">(</span>dup<span class="p">,</span> stack <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>panel.fun <span class="o">=</span> <span class="kr">function</span><span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> <span class="kc">...</span><span class="p">)</span> <span class="p">{</span>
        circos.genomicRect<span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> col <span class="o">=</span> <span class="s">&quot;red&quot;</span><span class="p">,</span>bg.border <span class="o">=</span> <span class="kc">NA</span><span class="p">,</span> cex<span class="o">=</span><span class="m">1</span> <span class="p">,</span> <span class="kc">...</span><span class="p">)</span>
<span class="p">})</span>
circos.genomicTrackPlotRegion<span class="p">(</span>del<span class="p">,</span> stack <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>panel.fun <span class="o">=</span> <span class="kr">function</span><span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> <span class="kc">...</span><span class="p">)</span> <span class="p">{</span>
        circos.genomicRect<span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> col <span class="o">=</span> <span class="s">&quot;blue&quot;</span><span class="p">,</span>bg.border <span class="o">=</span> <span class="kc">NA</span><span class="p">,</span> cex<span class="o">=</span><span class="m">1</span> <span class="p">,</span> <span class="kc">...</span><span class="p">)</span>
<span class="p">})</span>
</pre></div>
</td></tr></table>

<p>We can clearly see a massive deletion in chromosome 3.</p>
<p><br>
To finish we just need to draw 3 tracks + positional links to represent
SVs.</p>
<p>Unfortunately the vcf format has not been designed for SVs. SVs are
defined by 2 breakpoints and the vcf format stores the second one in the
info field. So we will need to extract this information to draw these
calls.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>chrEnd<span class="o">=</span><span class="kc">NULL</span>
posEnd<span class="o">=</span><span class="kc">NULL</span>
<span class="kr">for</span> <span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="kp">dim</span><span class="p">(</span>sv<span class="p">)[</span><span class="m">1</span><span class="p">])</span> <span class="p">{</span>
    addInfo<span class="o">=</span><span class="kp">strsplit</span><span class="p">(</span><span class="kp">as.character</span><span class="p">(</span>sv<span class="p">[</span>i<span class="p">,</span><span class="m">8</span><span class="p">]),</span>split<span class="o">=</span><span class="s">&quot;;&quot;</span><span class="p">)</span>
    chrInf<span class="o">=</span><span class="kp">strsplit</span><span class="p">(</span>addInfo<span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">3</span><span class="p">],</span>split<span class="o">=</span><span class="s">&quot;=&quot;</span><span class="p">)</span>
    chrEnd<span class="o">=</span><span class="kt">c</span><span class="p">(</span>chrEnd<span class="p">,</span>chrInf<span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">2</span><span class="p">])</span>
    posInf<span class="o">=</span><span class="kp">strsplit</span><span class="p">(</span>addInfo<span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">4</span><span class="p">],</span>split<span class="o">=</span><span class="s">&quot;=&quot;</span><span class="p">)</span>
    posEnd<span class="o">=</span><span class="kt">c</span><span class="p">(</span>posEnd<span class="p">,</span>posInf<span class="p">[[</span><span class="m">1</span><span class="p">]][</span><span class="m">2</span><span class="p">])</span>
<span class="p">}</span>
svTable<span class="o">=</span><span class="kt">data.frame</span><span class="p">(</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span>sv<span class="p">[,</span><span class="m">1</span><span class="p">],</span>sep<span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="kp">as.numeric</span><span class="p">(</span>sv<span class="p">[,</span><span class="m">2</span><span class="p">]),</span><span class="kp">as.numeric</span><span class="p">(</span>posEnd<span class="p">),</span><span class="kp">paste</span><span class="p">(</span><span class="s">&quot;chr&quot;</span><span class="p">,</span>chrEnd<span class="p">,</span>sep<span class="o">=</span><span class="s">&quot;&quot;</span><span class="p">),</span><span class="kp">as.character</span><span class="p">(</span>sv<span class="p">[,</span><span class="m">5</span><span class="p">]))</span>
</pre></div>
</td></tr></table>

<p><br>
Now that we have reformatted the SV calls, let’s draw them.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>typeE<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;&lt;DEL&gt;&quot;</span><span class="p">,</span><span class="s">&quot;&lt;INS&gt;&quot;</span><span class="p">,</span><span class="s">&quot;&lt;INV&gt;&quot;</span><span class="p">)</span>
colE<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;green&quot;</span><span class="p">)</span>
<span class="kr">for</span> <span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="m">3</span><span class="p">)</span> <span class="p">{</span>
        bed_list<span class="o">=</span>svTable<span class="p">[</span>svTable<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span>typeE<span class="p">[</span>i<span class="p">],]</span>
        circos.genomicTrackPlotRegion<span class="p">(</span>bed_list<span class="p">,</span>stack<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> panel.fun <span class="o">=</span> <span class="kr">function</span><span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> <span class="kc">...</span><span class="p">)</span> <span class="p">{</span>
                circos.genomicPoints<span class="p">(</span>region<span class="p">,</span> value<span class="p">,</span> cex <span class="o">=</span> <span class="m">0.5</span><span class="p">,</span> pch <span class="o">=</span> <span class="m">16</span><span class="p">,</span> col <span class="o">=</span> colE<span class="p">[</span>i<span class="p">],</span> <span class="kc">...</span><span class="p">)</span>
        <span class="p">})</span>
<span class="p">}</span>

bed1<span class="o">=</span><span class="kp">cbind</span><span class="p">(</span>svTable<span class="p">[</span>svTable<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">],</span>svTable<span class="p">[</span>svTable<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="m">2</span><span class="p">]</span><span class="m">+5</span><span class="p">)</span>
bed2<span class="o">=</span><span class="kp">cbind</span><span class="p">(</span>svTable<span class="p">[</span>svTable<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="kt">c</span><span class="p">(</span><span class="m">4</span><span class="p">,</span><span class="m">3</span><span class="p">)],</span>svTable<span class="p">[</span>svTable<span class="p">[,</span><span class="m">5</span><span class="p">]</span><span class="o">==</span><span class="s">&quot;&lt;TRA&gt;&quot;</span><span class="p">,</span><span class="m">3</span><span class="p">]</span><span class="m">+5</span><span class="p">)</span>

<span class="kr">for</span> <span class="p">(</span>i <span class="kr">in</span> <span class="m">1</span><span class="o">:</span><span class="kp">dim</span><span class="p">(</span>bed1<span class="p">)[</span><span class="m">1</span><span class="p">])</span> <span class="p">{</span>
    circos.link<span class="p">(</span>bed1<span class="p">[</span>i<span class="p">,</span><span class="m">1</span><span class="p">],</span>bed1<span class="p">[</span>i<span class="p">,</span><span class="m">2</span><span class="p">],</span>bed2<span class="p">[</span>i<span class="p">,</span><span class="m">1</span><span class="p">],</span>bed2<span class="p">[</span>i<span class="p">,</span><span class="m">2</span><span class="p">])</span>
<span class="p">}</span>
</pre></div>
</td></tr></table>

<p><br>
A good graph needs a title and legend:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>title<span class="p">(</span><span class="s">&quot;Somatic calls (SNV - SV - CNV)&quot;</span><span class="p">)</span>
legend<span class="p">(</span><span class="m">0.7</span><span class="p">,</span><span class="m">1.4</span><span class="p">,</span>legend<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;SNV&quot;</span><span class="p">,</span> <span class="s">&quot;CNV-DUPLICATION&quot;</span><span class="p">,</span><span class="s">&quot;CNV-DELETION&quot;</span><span class="p">,</span><span class="s">&quot;SV-DELETION&quot;</span><span class="p">,</span><span class="s">&quot;SV-INSERTION&quot;</span><span class="p">,</span><span class="s">&quot;SV-INVERSION&quot;</span><span class="p">),</span>col<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;orange&quot;</span><span class="p">,</span><span class="s">&quot;red&quot;</span><span class="p">,</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;blue&quot;</span><span class="p">,</span><span class="s">&quot;black&quot;</span><span class="p">,</span><span class="s">&quot;green&quot;</span><span class="p">,</span><span class="s">&quot;red&quot;</span><span class="p">),</span>pch<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">16</span><span class="p">,</span><span class="m">15</span><span class="p">,</span><span class="m">15</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">,</span><span class="m">16</span><span class="p">),</span>cex<span class="o">=</span><span class="m">0.75</span><span class="p">,</span>title<span class="o">=</span><span class="s">&quot;Tracks:&quot;</span><span class="p">,</span>bty<span class="o">=</span><span class="s">&#39;n&#39;</span><span class="p">)</span>
legend<span class="p">(</span><span class="m">0.6</span><span class="p">,</span><span class="m">0.95</span><span class="p">,</span>legend<span class="o">=</span><span class="s">&quot;SV-TRANSLOCATION&quot;</span><span class="p">,</span>col<span class="o">=</span><span class="s">&quot;black&quot;</span><span class="p">,</span>lty<span class="o">=</span><span class="m">1</span><span class="p">,</span>cex<span class="o">=</span><span class="m">0.75</span><span class="p">,</span>lwd<span class="o">=</span><span class="m">1.2</span><span class="p">,</span>bty<span class="o">=</span><span class="s">&#39;n&#39;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p><br>
You should obtain a plot like this one:</p>
<p><img alt="image" src="../images/circos.png" /></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Could you generate the graph and save it into a pdf file?</p>
<div class="admonition success"><details class="spoilers"><summary><strong>Answer</strong></summary><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>circos.clear<span class="p">()</span>
pdf<span class="p">(</span><span class="s">&quot;circos.pdf&quot;</span><span class="p">)</span>
<span class="kc">...</span>
dev.off<span class="p">()</span>
</pre></div>
</td></tr></table>
</details></div>
</div>
<p>Finally exit R
  <table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">q</span><span class="p">(</span><span class="s">&quot;yes&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table></p>
<hr />
<h2 id="acknowledgements">Acknowledgements<a class="headerlink" href="#acknowledgements" title="Permanent link">&para;</a></h2>
<p>Mathieu Bourgey would like to thank and acknowledge Louis Letourneau for this help and
for sharing his material. The format of the tutorial has been inspired
from Mar Gonzalez Porta. I also want to acknowledge Joel Fillon, Louis
Letrouneau (again), Francois Lefebvre, Maxime Caron and Guillaume
Bourque for the help in building these pipelines and working with all
the various datasets.</p>
<h2 id="license">License<a class="headerlink" href="#license" title="Permanent link">&para;</a></h2>
<p>This work is licensed under a Creative Commons Attribution-ShareAlike
3.0 Unported License. This means that you are able to copy, share and
modify the work, as long as the result is distributed under the same
license.</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Structural Variant Analysis
              </span>
            </div>
          </a>
        
        
          <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Assessing somatic mutational signatures
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-6b4ff16fa7.js"></script>
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/spoilers.js"></script>
      
    
    
      
    
  </body>
</html>