
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.1">
    
    
      
        <title>Cohort analysis for the identification of driver genes - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-f98ca285f9.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Cancer Genomics Modules
                </span>
              
                <span class="md-header-nav__parent">
                  Somatic Mutations
                </span>
              
            
            Cohort analysis for the identification of driver genes
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link md-tabs__link--active">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6" checked>
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Cohort analysis for the identification of driver genes
      </label>
    
    <a href="./" title="Cohort analysis for the identification of driver genes" class="md-nav__link md-nav__link--active">
      Cohort analysis for the identification of driver genes
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysing-cancer-cohort-data-with-intogen" title="Analysing cancer cohort data with IntOGen" class="md-nav__link">
    Analysing cancer cohort data with IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exploring-the-output-of-intogen" title="Exploring the output of IntOGen" class="md-nav__link">
    Exploring the output of IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#analysing-cancer-cohort-data-with-intogen" title="Analysing cancer cohort data with IntOGen" class="md-nav__link">
    Analysing cancer cohort data with IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#exploring-the-output-of-intogen" title="Exploring the output of IntOGen" class="md-nav__link">
    Exploring the output of IntOGen
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Cohort analysis for the identification of driver genes</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Run the <code class="codehilite">IntOGen</code> analysis software on cohort mutation data.</p>
</li>
<li>
<p>Have gained experience of the structure of the analysis output files
    in order to identify potential driver genes.</p>
</li>
<li>
<p>Have gained overview knowledge of different methods for
    identification of genes important in cancers.</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>IntOGen mutations platform:<br />
<a href="https://www.intogen.org/search">https://www.intogen.org/search</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p>TCGA melanoma somatic SNV data from 338 tumour samples:<br />
<a href="https://tcga-data.nci.nih.gov/tcga/">https://tcga-data.nci.nih.gov/tcga/</a></p>
<h3 id="useful-links">Useful Links<a class="headerlink" href="#useful-links" title="Permanent link">&para;</a></h3>
<p>Mutation Annotation Format (MAF) specification:<br />
<a href="https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification">https://wiki.nci.nih.gov/display/TCGA/Mutation+Annotation+Format+(MAF)+Specification</a></p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>Cancer driver genes are commonly described as genes that when mutated
directly affect the potential of a cell to become cancerous. They are
important to a tumour cell as they confer a growth or survival advantage
over the normal surrounding cells. The mutations in these driver genes
are then clonally selected for as the population of tumour cells
increases. We think of the key genes driving tumour initiation
(development), progression, metastases, resistance and survival. Driver
gene mutations are often described as &ldquo;early&rdquo; events because they were
key in turning a normally functioning and regulated cell into a
dysregulated one. The logical assumption is that these key mutations
will be present in all tumour cells in a patient’s sample; although
sometime this is not true.</p>
<p>There are two major research goals that underline the need to identify
driver genes:</p>
<ul>
<li>
<p>By identifying the early changes that take place researchers might
    be able to find a treatment to stop the root cause of why cells
    become malignant.</p>
</li>
<li>
<p>By identifying groups of patients with the same genes mutated then
    we can develop therapies that will work for all of them.</p>
</li>
</ul>
<p>When we sequence tumour samples we tend to use samples that come from
fully developed cancers that can carry hundreds to thousands of
mutations in genes and many more outside of genes. The accumulation of
these passenger mutations in cancer cells can happen because often the
repair mechanisms or damage sensing processes are amongst the first
pathways to become disrupted accelerating the mutational rate. Mutations
that occur in genes after the cell has become cancerous may still affect
the growth rate, invasiveness and even the response to chemotherapy but
may not be present in all cells of a tumour. These genes may be drivers
of chemo-resistance or metastasis and are equally good targets for
therapies.</p>
<p>IntOGen-mutations is a platform that aims to identify driver mutations
using two methodologies from cancer cohort mutation data: the first
identifies mutations that are most likely to have a functional impact
combined with identifying genes that are frequently mutated; and the
second, genes that harbour clustered mutations. These measures are all
indicators of positive selection that occurs in cancer evolution and may
help the identification of driver genes.</p>
<hr />
<h2 id="analysing-cancer-cohort-data-with-intogen">Analysing cancer cohort data with IntOGen<a class="headerlink" href="#analysing-cancer-cohort-data-with-intogen" title="Permanent link">&para;</a></h2>
<p>IntOGen-mutations is a web platform that can allow users to run their
analysis on the host’s servers or it can be downloaded and run without
any limits on the number of analyses on a local server.</p>
<p>For the purposes of the course we will be using a local version of
IntOGen so that we don’t encounter any issues sharing resources.</p>
<ul>
<li>
<p>To start IntOGen open a terminal and navigate to the directory <code class="codehilite">somatic/intogen_mutations_analysis</code>.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nb">cd</span> ~/somatic/intogen_mutations_analysis
</pre></div>
</td></tr></table>

</li>
<li>
<p>Start <code class="codehilite">IntOGen</code> by typing</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>./run web
</pre></div>
</td></tr></table>

</li>
</ul>
<p>A browser window should open up with the <code class="codehilite">IntOGen</code> home page displayed.
  <code class="codehilite">IntoGen</code> requires you to log in to Mozilla persona. On the top right hand side
  of the screen there will be a <code class="codehilite">Sign in</code> button.</p>
<ul>
<li>Click the <code class="codehilite">Sign in</code> button.</li>
</ul>
<p>If you can access your email from the web you can create yourself an
  account. You do need to verify your email address by logging in to your
  webmail. Alternatively, you can log in using these dummy account details.</p>
<blockquote>
<p>Email address: <a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#97;&#109;&#112;&#46;&#103;&#101;&#110;&#111;&#109;&#105;&#99;&#115;&#64;&#103;&#109;&#97;&#105;&#108;&#46;&#99;&#111;&#109;">&#97;&#109;&#112;&#46;&#103;&#101;&#110;&#111;&#109;&#105;&#99;&#115;&#64;&#103;&#109;&#97;&#105;&#108;&#46;&#99;&#111;&#109;</a>.<br />
Password if required: letmepass.</p>
</blockquote>
<ul>
<li>
<p>Click on the <code class="codehilite">Analysis</code> button on the top tool bar. Click on the
<code class="codehilite">Analyse your data</code> button in the <code class="codehilite">Cohort analysis</code> panel.</p>
</li>
<li>
<p>A form will open up, click on the <code class="codehilite">Load</code> button and navigate to the
<code class="codehilite">TCGA_Melanoma_SMgene.maf</code> file located in the directory path
<code class="codehilite">somatic/intogen/TCGA_Melanoma_SMgene.maf</code>.</p>
</li>
<li>
<p>Ensure the Genome assembly is set to <code class="codehilite">hg19(GRCh37)</code>.</p>
</li>
<li>
<p>The click the <code class="codehilite">Start analysis</code> button.</p>
</li>
</ul>
<p><br>
The TCGA melanoma maf used in this practical has been modified from the
original to reduce processing time and only contains data for the top
680 mutated genes.</p>
<p>The tool will take around 10 minutes to run and the progress will be
indicated by the green bar. You may need to refresh the browser if it
looks like there has been no progress for a while. Once complete the output can
be downloaded.</p>
<p><br></p>
<ul>
<li>
<p>Click on the download link to the left of the green bar or go through
the results button at the top of the page.</p>
</li>
<li>
<p>Select to <code class="codehilite">save</code> the file which will end up in the download directory
<code class="codehilite">/Downloads</code>.</p>
</li>
<li>
<p>Click on the blue download arrow at the top right hand side of the
browser window.</p>
</li>
<li>
<p>Click on the file called <code class="codehilite">TCGA_Melanoma_slimSMgene.zip</code>.</p>
</li>
<li>
<p>Highlight the file name in the pop-up window and click on <code class="codehilite">Extract</code>.</p>
</li>
<li>
<p>Select the <code class="codehilite">/trainee/somatic/intogen/</code> directory and click on the
<code class="codehilite">Extract</code> button at the bottom of the window.</p>
</li>
<li>
<p>Click to show files once extracted.</p>
</li>
<li>
<p>Double click on the directory called <code class="codehilite">TCGA_Melanoma_slimSMgene</code>.</p>
</li>
</ul>
<hr />
<h2 id="exploring-the-output-of-intogen">Exploring the output of IntOGen<a class="headerlink" href="#exploring-the-output-of-intogen" title="Permanent link">&para;</a></h2>
<p>When you run your data over the web on the remote site there is a browse
facility that allows you to explore your data using the web version of
the database. Running <code class="codehilite">IntOGen</code> locally provides the same tabular
information but in a flat file format.</p>
<p>There are 7 files generated in a successful run of <code class="codehilite">IntOGen</code>:</p>
<ul>
<li>
<p><code class="codehilite">genes.tsv</code> - this is the main output summary table.</p>
</li>
<li>
<p><code class="codehilite">variant_genes.tsv</code> - describes the variants identified in the genes</p>
</li>
<li>
<p><code class="codehilite">variant_samples.tsv</code> - lists the variants together with the samples
    Ids</p>
</li>
<li>
<p><code class="codehilite">pathways.tsv</code> - lists the Kegg pathway ID for perturbed pathways</p>
</li>
<li>
<p><code class="codehilite">consequences.tsv</code> - lists consequences of the variants in all known
    transcripts</p>
</li>
<li>
<p><code class="codehilite">project.tsv</code> - one line to summarise the input data</p>
</li>
<li>
<p><code class="codehilite">fimpact.gitools.tdm</code> - the output of the functional impact tool</p>
</li>
</ul>
<p><br>
The scope of this practical will concentrate on identification of driver
genes so we will look at the main output concerning genes. <code class="codehilite">OncodriveFM</code>
does however also calculates a functional impact bias of high impact
mutations in annotated <a href="http://www.genome.jp/kegg/pathway.html">KEGG pathways</a>
as shown in the <code class="codehilite">pathways.tsv</code> file but will not be used in this course.</p>
<ul>
<li>Open up the <code class="codehilite">genes.tsv</code> in the spreadsheet software by double clicking
on the file.</li>
</ul>
<p>This file contains the overall summary results for the <code class="codehilite">IntOGen</code> pipeline
presented by gene and reports Q values (i.e. multiple testing corrected
P values) for the mutation frequency and cluster modules.</p>
<p>Significantly mutated genes from the cohort data are identified using
the <code class="codehilite">OncodriveFM</code> module of <code class="codehilite">IntOGen</code>. This tool detects genes that have
accumulated mutations with a high functional impact. It uses annotations
from the Ensembl variant effect predictor (VEP, V.70) that includes SIFT
and Polyphen2 and precomputed MutationAssessor functional impacts. It
calculates a P value per gene from the number of mutations detected
across all possible coding bases of a gene with a positive weighting for
mutations with a high functional impact.</p>
<ul>
<li>Sort the data in this file by two levels starting with the 5<sup>th</sup> column,
the FM_QVALUE from smallest-to-biggest and then by the 12<sup>th</sup> column the
CLUST_QVALUE also from smallest-to-biggest.</li>
</ul>
<p>The top nine genes with the smallest Q values should be TP53, PTEN,
PPP6C, CDKN2A, BRAF, NRAS, ARID2, TTN, IDH1.</p>
<p>All of these have very small FM_Qvalues which means they are all
significantly mutated genes in this TCGA Melanoma cohort of 338
patients.</p>
<ul>
<li>Now look at their sample frequency (column 9 SAMPLE_FREQ) values these
are the number of samples that contain at least one mutation in the
gene.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>a)   ​Which gene has mutations in the most samples?</p>
<p>b)   ​Which gene had the lowest FM Q value?</p>
<p>c)   Why don’t the genes with the lowest Q values also have the
highest sample frequency value?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>a)  TTN has the highest number of samples with mutations. There are
    265 out of 327 samples with mutations in TTN.</p><p>b)  TP53 or PTEN</p><p>c)  ​The P value calculation takes into account the length of the
    coding sequence of the gene, the mutation rate of the nucleotides
    contained within it and the functional consequences of those
    changes. Therefore a small gene with a small number of deleterious
    mutations may have a lower P value and also Q value than a large
    gene with a high mutation frequency.</p></details></div>
</div>
<p>The results for the assessment of clustered mutations in genes carried
out by the <code class="codehilite">OncodriveCLUST</code> module of <code class="codehilite">IntOGen</code> are shown in the 10-14<sup>th</sup>
columns. The Q value is in column 12, CLUST_QVALUE which indicates if
there is a significant grouping of mutations identified. The positions
of the mutations are now described in terms of the amino acid residue in
the encoded protein.</p>
<p>The three oncogenes BRAF, NRAS and IDH1 have very low CLUST_QVALUEs
&lt;0.01 indicating that the mutations in these genes are highly
clustered. The CLUST_COORDS column reports that there are 158 samples
with mutations between the amino acid positions 594-601 of BRAF; 84
samples with mutations at amino acid position 61 of NRAS; and 15 sample
with mutations at amino acid position 132 of IDH1.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Why are the oncogenes more likely to have clustered mutations and the
tumour suppressor genes less likely?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Gain of function mutations are required to activate oncogenes and so
only key residues in the protein will result in activation. Tumour
suppressors are frequently affected by loss of function mutations and
deletions. A truncating mutation or frameshift indel can occur in any
exon, except the last one, and have the same deleterious functional
result.</p></details></div>
</div>
<p>The INTOGEN_DRIVER column indicates if this gene is a known cancer
driver gene with 1 for yes and 0 for no so it is promising to see the
majority of our top hit genes are known drivers.</p>
<p>The XREFS column indicates a mutation match in external databases. This
means the position and the base change has been seen before. For the
known Driver genes there are many COSMIC IDs indicating these mutations
have been recorded in the <a href="http://cancer.sanger.ac.uk/cosmic">Catalogue of Somatic Mutations in Cancer database</a>.
Remember the majority of data in these databases have come from large scale cancer sequencing
projects carried out by TCGA and ICGC associated groups.</p>
<ul>
<li>Have a look at the values in the XREFS field for the gene TTN.</li>
</ul>
<p>There are a large number of COSMIC entries for mutations in this gene
but also a large number of ESP ID numbers that refer to the National
Heart, Lung, and Blood Institute (NHLBI), <a href="http://evs.gs.washington.edu/EVS/">Exome Sequencing Project (ESP)</a>.</p>
<p>This data comes from a diverse population that typically don’t have
cancer but focus on patients with heart, lung and blood disorders.</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can we draw any conclusions from the high number of ESP mutations and
dbSNP references for TTN?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Functional studies would be the only way to prove conclusively if TTN
mutations were cancer driver mutations. The high number of samples in
the ESP cohort that have mutations in common with the cancer cohorts
could indicate that TTN mutation may not be unique to the cancer
setting. This conclusion seems to be backed up by the high number of
mutations also with dbSNP ids that indicates the potential for some of
these mutations to be present in the general population.</p><p>Amongst this high number of mutations there may be groups of patients
where the mutations are purely passenger and other groups where the
mutations could contribute to the tumour.</p><p>From this data there is no way to tell.</p></details></div>
</div>
<p><br>
The other files in the output support the information in this sheet.</p>
<p>The <code class="codehilite">variant_genes.tsv</code> file includes a summary of all mutations found in
each of the genes with a count of the number of samples identified to
have that mutation. It also reports the variant impact score and
category of which there are four; high, medium, low and none.</p>
<ul>
<li>Open up the <code class="codehilite">variant_genes.tsv</code> file and explore the data.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can you find out what the nucleotide change details for the most common
BRAF mutation that results in V600E amino acid change in the cohort?
Sort the data on the gene symbol column to make this easier.</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>It is an A&gt;T at position chr7:140453136 identified in 127 samples.</p></details></div>
</div>
<p><br>
The <code class="codehilite">variant_samples.tsv</code> worksheet allows you to find out the sample
identification numbers for the samples with the mutation that you are
interested in.</p>
<ul>
<li>Open up the <code class="codehilite">variant_samples.tsv</code> file and explore the data.</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can you use the position that you found in the question above to locate
the sample IDs for the 3 cases with a V600R BRAF mutation that is caused
by a nucleotide change of AC/CT starting at chr7:140453136?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>TCGA-D3-A1QA-06A-11D-A196-08, TCGA-EB-A3XC-01A-11D-A23B-08, TCGA-ER-A193-06A-12D-A197-08</p></details></div>
</div>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<p>Gunes et al. Nat. Methods 2010
:   <a href="http://www.nature.com/nmeth/journal/v7/n2/pdf/nmeth0210-92.pdf">http://www.nature.com/nmeth/journal/v7/n2/pdf/nmeth0210-92.pdf</a></p>
<p>Gonzalez-Perez et al. Nat. Methods 2013
:   <a href="http://www.nature.com/nmeth/journal/v10/n11/pdf/nmeth.2642.pdf">http://www.nature.com/nmeth/journal/v10/n11/pdf/nmeth.2642.pdf</a></p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../01_signatures/" title="Assessing somatic mutational signatures" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Assessing somatic mutational signatures
              </span>
            </div>
          </a>
        
        
          <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                DeNovo Command Line
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-e3caa82af6.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>