
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.1">
    
    
      
        <title>Bacterial Assembly - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-f98ca285f9.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  DeNovo Modules
                </span>
              
            
            Bacterial Assembly
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link md-tabs__link--active">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" checked>
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
    <a href="./" title="Bacterial Assembly" class="md-nav__link md-nav__link--active">
      Bacterial Assembly
    </a>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h1>
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Perform a simple genome assembly for a small organism using <code class="codehilite">Velvet</code></p>
</li>
<li>
<p>Visualise the assembly graph using <code class="codehilite">Bandage</code></p>
</li>
<li>
<p>Be aware of the effects and trade-offs of the parameter <code class="codehilite">K</code> on the
    genome assembly</p>
</li>
<li>
<p>Understand that genome assembly produces a graph structure</p>
</li>
</ul>
<h1 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h1>
<h2 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h2>
<p>Velvet:<br />
<a href="https://www.ebi.ac.uk/~zerbino/velvet/">https://www.ebi.ac.uk/~zerbino/velvet/</a></p>
<p>Bandage:<br />
<a href="https://rrwick.github.io/Bandage/">https://rrwick.github.io/Bandage/</a></p>
<h1 id="data-source-and-useful-links">Data Source and Useful Links<a class="headerlink" href="#data-source-and-useful-links" title="Permanent link">&para;</a></h1>
<p>Data used in this tutorial is from ENA accession SRR2054105<br />
<a href="http://www.ebi.ac.uk/ena/data/view/SRR2054105">http://www.ebi.ac.uk/ena/data/view/SRR2054105</a></p>
<p>Spreadsheet used in the group activity:<br />
<a href="https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing">https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing</a>&gt;</p>
<h1 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h1>
<p>In this tutorial we will take raw sequencing reads and <em>de novo</em>
assemble them into contigs. We will also explore the internal assembly
graph structure to aid our understanding of why assemblies are
incomplete.</p>
<h1 id="before-the-assembly">Before the assembly<a class="headerlink" href="#before-the-assembly" title="Permanent link">&para;</a></h1>
<h2 id="sequence-data">Sequence data<a class="headerlink" href="#sequence-data" title="Permanent link">&para;</a></h2>
<p>We have sequenced the genome of a bacterium using paired-end chemistry
on an Illumina HiSeq 2000 instrument at the University of Oxford and
publicly available as SRR2054105 the Europen Nucleotide Archive (ENA).</p>
<p>Let’s have a look at the datasets by doing:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span> cd /home/trainee/bacterial
 ls
</pre></div>
</td></tr></table>

<p>You should see the following two files:</p>
<p>R1.fastq.gz</p>
<p>R2.fastq.gz</p>
<p>How many reads are in this data set?\</p>
<p>2100778 pairs</p>
<p>What is the yield in basepairs?\</p>
<p>210077800 bp = 210 Mbp</p>
<p>Assuming an average bacterial genome size of 4 Mbp, what depth of
coverage do we have?\</p>
<p>210 / 4 = 53x</p>
<h2 id="trim-or-clip-reads">Trim or clip reads<a class="headerlink" href="#trim-or-clip-reads" title="Permanent link">&para;</a></h2>
<p>There are two distinct reasons one may wish to trim or clip the raw
reads.</p>
<ol>
<li>
<p>Low quality bases typically occur toward the end of Illumina reads.
    The lower the quality score, the higher the chance that the base is
    incorrect. This introduces false k-mers into the assembly process. A
    good assembler should handle these gracefully.</p>
</li>
<li>
<p>Sequencing adaptors are artificial sequence that can occur at the
    end of reads that came from fragments of DNA that were shorter than
    desired. They were not in the original genome and their presence in
    lots of reads will totally confuse the assembler.</p>
</li>
</ol>
<p>Normally one would use <code class="codehilite">fastqc</code> or similar to determine whether quality
trimming is warranted and for the presence of sequencing adaptors. The
data set in this exercise is 99% free of adaptors and is good quality
overall so we will skip those steps.</p>
<p>How do you know which adaptor sequence to trim?\</p>
<p>You either ask the sequencing provider or use a tool that tries all
known adapters.</p>
<p>How are are quality values in <code class="codehilite">FASTQ</code> files calculated? Can we trust
them?\</p>
<p>The are calculated using formulas calibrated to the physical
measurements the sequencing instrument takes. In the case of Illumina
sequencing, these measurements will include pixel light intensities from
the images the camera/scanner takes of the flow cell.</p>
<h2 id="allocate-yourself-to-a-k-value-on-the-spreadsheet">Allocate yourself to a <code class="codehilite">K</code> value on the spreadsheet<a class="headerlink" href="#allocate-yourself-to-a-k-value-on-the-spreadsheet" title="Permanent link">&para;</a></h2>
<p>An important parameter to most genome assemblers is <code class="codehilite">K</code> which is the
k-mer size used to construct the <em>de Bruijn</em> graph (pronounced <em>de
Brown</em>) .</p>
<p>Please go to this shared online
spreadsheet <a href="https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing">https://docs.google.com/spreadsheets/d/1iFbCCihawpY1LClsAB-OJ66lyeW7EsdJyaMo1HCetF8/edit?usp=sharing</a>
and choose <strong>a K value</strong> and put your name next to it.</p>
<h1 id="assemble-the-reads-using-velvet">Assemble the reads using Velvet<a class="headerlink" href="#assemble-the-reads-using-velvet" title="Permanent link">&para;</a></h1>
<p>The <code class="codehilite">Velvet</code> software performs the assembly in two steps. The first
<code class="codehilite">velveth</code> step hashes the reads. The second <code class="codehilite">velvetg</code> step builds,
cleans and traverses the graph.</p>
<p>Choose an output folder <code class="codehilite">DIR</code> and use the <code class="codehilite">K</code> value you were allocated
on the spreadsheet and assemble the reads.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>velveth DIR K -shortPaired -fastq.gz -separate R1.fastq.gz R2.fastq.gz
</pre></div>
</td></tr></table>

<p>Description of the arguments used in the command:</p>
<p>DIR
:   = Directory name you chose to write the output to</p>
<p>K
:   = K-mer value; it must be an odd number</p>
<p>-shortPaired
:   = The read types are short and paired-end</p>
<p>-fastq.gz
:   = The format of the input files</p>
<p>-separate
:   = R1 and R2 reads are in separate files</p>
<p>input file names
:   = input FASTQ file names</p>
<p>Run the velvetg step now:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>/usr/bin/time -f &quot;%e&quot; velvetg DIR -exp_cov auto -cov_cutoff auto
</pre></div>
</td></tr></table>

<p>Description of the arguments used in the command:</p>
<p>time
:   = to capture the <code class="codehilite">velvetg</code> command run time</p>
<p>DIR
:   = Directory name you chose to write the velveth output to</p>
<p>-exp_cov
:   = expected coverage is set to <code class="codehilite">auto</code>. To be determined by <code class="codehilite">velvet</code></p>
<p>-cov_cutoff
:   = coverage cutoff is set to <code class="codehilite">auto</code>. To be determined by <code class="codehilite">velvet</code></p>
<h2 id="add-your-results-to-the-spreadsheet">Add your results to the spreadsheet<a class="headerlink" href="#add-your-results-to-the-spreadsheet" title="Permanent link">&para;</a></h2>
<p>[H]</p>
<p>ll</p>
<p><strong>Column</strong> &amp; <strong>Where to find the value</strong>\
<code class="codehilite">K-mer size</code> &amp; You chose this earlier and used it in the velveth
command\
<code class="codehilite">How long it took to run</code> &amp; Look at the final output line for the user
time in seconds: NN.N\
<code class="codehilite">Average K-mer coverage</code> &amp; Look for Estimated Coverage = NN.N\
<code class="codehilite">Number of contigs</code> &amp; Look for text Final graph has NNN nodes\
<code class="codehilite">N50 contig size</code> &amp; Look for n50 of NNNNN\
<code class="codehilite">Largest contig size</code> &amp; Look for max NNNNN\
<code class="codehilite">Total number of basepairs (sum of contig lengths)</code> &amp; Look for total
NNNNNNN\</p>
<p>[tab:results]</p>
<h2 id="effect-of-k-on-assembly">Effect of K on assembly<a class="headerlink" href="#effect-of-k-on-assembly" title="Permanent link">&para;</a></h2>
<p>Examine the spreadsheet and look for patterns in the tabular data and
corresponding bar/line charts.</p>
<p>How does K affect each of the statistics? Which value of K do you think
is doing the best job? Why?</p>
<h2 id="output-files">Output Files<a class="headerlink" href="#output-files" title="Permanent link">&para;</a></h2>
<p>[H]</p>
<p>ll</p>
<p><strong>Filename</strong> &amp; <strong>Description</strong>\
<code class="codehilite">contigs.fa</code> &amp; this is a FASTA file with your contigs\
<code class="codehilite">Log</code> &amp; has most of the metrics in it that we recorded\
<code class="codehilite">stats.txt</code> &amp; TSV file of length and coverage of individual contigs\
<code class="codehilite">Sequences</code> &amp; A copy of the input FASTQ sequences in FASTA format\
<code class="codehilite">Pregraph Roadmaps Graph2</code> &amp; Interim assembly graph structure\
<code class="codehilite">LastGraph</code> &amp; Final graph structure\</p>
<p>[tab:velvet<sub>o</sub>ut]</p>
<p>Let’s examine the <code class="codehilite">stats.txt</code> file and look at the <code class="codehilite">short1_cov</code> column
which is the k-mer coverage of each contig:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cut -f6 dir/stats.txt | less
</pre></div>
</td></tr></table>

<p>Press <code class="codehilite">SPACE</code> and <code class="codehilite">b</code> to scroll up and down, and press <code class="codehilite">q</code> to quote.</p>
<p>What do you notice about the k-mer coverages?\</p>
<p>They all appear to have a similar value, but there are some that do not
fit the pattern.</p>
<p>What do the outliers correspond to?\</p>
<p>Repetitive elements of the genome including gene duplications. Also
replicons with differing copy numbers, like bacterial plasmids.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>grep NN dir/contigs.fa
</pre></div>
</td></tr></table>

<p>Why is there N letters in the assembly?\</p>
<p>Paired-end reads come from each end of the same original fragment of
DNA. The unsequenced gap in the middle is variable, and unknown. The
assembler sometimes struggles to resolve these gaps. So it knows two
sections of the genome are connected, but it doesn’t quite know what is
between them. So it pads them with its best guess of how many bases
there are, and uses N to denote them as unknown.</p>
<h1 id="visualising-the-assembly-graph-using-bandage">Visualising the assembly graph using Bandage<a class="headerlink" href="#visualising-the-assembly-graph-using-bandage" title="Permanent link">&para;</a></h1>
<p>The final graph that <code class="codehilite">Velvet</code> uses is stored in the file <code class="codehilite">LastGraph</code>.
The <code class="codehilite">Bandage</code> software allows us to view and explore the assembly graph.</p>
<ol>
<li>
<p>Start <code class="codehilite">Bandage</code>.</p>
</li>
<li>
<p>Go to <code class="codehilite">File -&gt; Load Graph</code> and load the <code class="codehilite">LastGraph</code> from your
    assembly in <code class="codehilite">DIR</code>. This may take a little while so be patient.</p>
</li>
<li>
<p>Maximize your window to fill up the whole screen.</p>
</li>
<li>
<p>Click <code class="codehilite">Draw graph</code> on the left hand panel.</p>
</li>
<li>
<p>Change <code class="codehilite">Random colours</code> to <code class="codehilite">Colour by read depth</code> on the left hand
    panel.</p>
</li>
</ol>
<p>Now get up out of your chair and walks around and look at all the
different graphs the other participants obtained with different values
of <code class="codehilite">K</code>.</p>
<p>How does <code class="codehilite">K</code> affect the graph?\</p>
<p>Small K produces higher k-mer coverage but also more connectivity as
smaller K is more ambiguous.</p>
<p>What would the graph look like in an ideal situtation?\</p>
<p>If the genome had M replicons, we would like to see only M sub-graphs.
Each sub-graph would be linear or circular, depending on the form of the
replicon in the original organism.</p>
<p>Why didn’t anyone achieve perfection?\</p>
<p>Short reads are unable to disambiguate repeats longer than the read
length (or read span). Most genomes have repeats beyond 500 bp.</p>
<p>Here is another example
<a href="https://github.com/rrwick/Bandage/wiki/Effect-of-kmer-size">https://github.com/rrwick/Bandage/wiki/Effect-of-kmer-size</a> from the
Bandage web site on how K can affect assembly graphs.</p>
<h2 id="explore-the-graph">Explore the graph<a class="headerlink" href="#explore-the-graph" title="Permanent link">&para;</a></h2>
<p><code class="codehilite">Bandage</code> is designed to be an interactive tool. It allows you to see
relationships between parts of your genome that are lost when you only
look at the FASTA file of contigs.</p>
<ul>
<li>
<p>Zoom in using the <code class="codehilite">Zoom</code> up/down button in the left hand panel</p>
</li>
<li>
<p>Pan around by holding down the Option/Windows key and dragging on
    the background</p>
</li>
<li>
<p>Move nodes out of the way by selecting and dragging</p>
</li>
</ul>
<p>Feel free to play around a bit and ask questions.</p>
<h2 id="features-of-the-assembly-graph">Features of the assembly graph<a class="headerlink" href="#features-of-the-assembly-graph" title="Permanent link">&para;</a></h2>
<p>The graph is quite tangled. The long stretches correspond to contigs.
The intersections correspond to shared k-mers in the graph, which occur
due to repeated sequences within the genome.</p>
<p>Select an node (rectangle) in the graph. It’s length is reported in the
right hand panel as <code class="codehilite"><span class="n">Length</span><span class="o">:</span> <span class="n">NNN</span></code>.</p>
<p>On the menu choose <code class="codehilite">Output -&gt; Web BLAST selected node</code>. Your browser
should open the NCBI BLAST Website
<a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&amp;PAGE_TYPE=BlastSearch&amp;LINK_LOC=blasthome">http://blast.ncbi.nlm.nih.gov/Blast.cgi?PROGRAM=blastn&amp;PAGE_TYPE=BlastSearch&amp;LINK_LOC=blasthome</a>.</p>
<p>Click the <code class="codehilite">BLAST</code> button at the bottom, and wait for the result.</p>
<p>Did your node match anything in the Genbank database?\</p>
<p>You will need to examine the BLAST report on the trainee browser window.</p>
<p>Can you determine what species of bacteria was sequenced?\</p>
<p>The top hit of the small segment of DNA the trainee chose may not
necessarily reflect the true species of bacteria. But if multiple
segments produce consistent top hits to the same species you would have
some confidence.</p>
<h1 id="conclusion">Conclusion<a class="headerlink" href="#conclusion" title="Permanent link">&para;</a></h1>
<p>You should now:</p>
<ul>
<li>
<p>know how to use Velvet to assemble a simpel genome from Illumina
    sequences</p>
</li>
<li>
<p>understand the role of the k-mer length K in the assembly process</p>
</li>
<li>
<p>be able to relate the graph structure to the final contigs</p>
</li>
<li>
<p>realize the limitations of short read sequences with respect to
    genome complexity</p>
</li>
</ul>
<p>Thank you.</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                DeNovo Command Line
              </span>
            </div>
          </a>
        
        
          <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Eukaryote Assembly
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-e3caa82af6.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>