
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.1">
    
    
      
        <title>Copy Number Variation - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-f98ca285f9.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Cancer Genomics Modules
                </span>
              
            
            Copy Number Variation
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link md-tabs__link--active">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Copy Number Variation
      </label>
    
    <a href="./" title="Copy Number Variation" class="md-nav__link md-nav__link--active">
      Copy Number Variation
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequenza-cnv-analysis" title="Sequenza CNV Analysis" class="md-nav__link">
    Sequenza CNV Analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-pre-processing-walking-the-genome" title="Step 1: Pre-Processing – Walking the Genome" class="md-nav__link">
    Step 1: Pre-Processing – Walking the Genome
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-perform-binning" title="Step 2: Perform Binning" class="md-nav__link">
    Step 2: Perform Binning
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-running-sequenza-algorithms-and-plotting-results" title="Step 3: Running Sequenza Algorithms and Plotting Results" class="md-nav__link">
    Step 3: Running Sequenza Algorithms and Plotting Results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequenza-analysis-results-and-visualisation" title="Sequenza Analysis Results and Visualisation" class="md-nav__link">
    Sequenza Analysis Results and Visualisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cnv-visualisationconfirmation-in-igv" title="CNV Visualisation/Confirmation in IGV" class="md-nav__link">
    CNV Visualisation/Confirmation in IGV
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequenza-cnv-analysis" title="Sequenza CNV Analysis" class="md-nav__link">
    Sequenza CNV Analysis
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#step-1-pre-processing-walking-the-genome" title="Step 1: Pre-Processing – Walking the Genome" class="md-nav__link">
    Step 1: Pre-Processing – Walking the Genome
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-2-perform-binning" title="Step 2: Perform Binning" class="md-nav__link">
    Step 2: Perform Binning
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#step-3-running-sequenza-algorithms-and-plotting-results" title="Step 3: Running Sequenza Algorithms and Plotting Results" class="md-nav__link">
    Step 3: Running Sequenza Algorithms and Plotting Results
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sequenza-analysis-results-and-visualisation" title="Sequenza Analysis Results and Visualisation" class="md-nav__link">
    Sequenza Analysis Results and Visualisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cnv-visualisationconfirmation-in-igv" title="CNV Visualisation/Confirmation in IGV" class="md-nav__link">
    CNV Visualisation/Confirmation in IGV
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Copy Number Variation</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Understand and perform a simple copy number variation analysis on
    NGS data</p>
</li>
<li>
<p>Become familiar with Sequenza</p>
</li>
<li>
<p>Understand the CNV inference process as an interplay between depth
    of sequencing, cellularity and B-allele frequency</p>
</li>
<li>
<p>Visualize CNV events by manual inspection</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Sequenza:<br />
<a href="http://www.cbs.dtu.dk/biotools/sequenza/">http://www.cbs.dtu.dk/biotools/sequenza/</a></p>
<p>IGV:<br />
<a href="http://www.broadinstitute.org/igv/">http://www.broadinstitute.org/igv/</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p>Raw data download:<br />
<a href="http://sra.dnanexus.com/studies/ERP001071">http://sra.dnanexus.com/studies/ERP001071</a></p>
<p>Data publication:<br />
<a href="http://www.ncbi.nlm.nih.gov/pubmed/22194472">http://www.ncbi.nlm.nih.gov/pubmed/22194472</a></p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>The goal of this hands-on session is to perform a copy number variation
analysis (CNV) on a normal/tumour pair of alignment files (BAMs)
produced by the mapping of Illumina short read sequencing data.</p>
<p>To ensure reasonable analysis times, we will perform the analysis on a
heavily subset pair of BAM files. These files contain just the first
60Mb of chromosome 5 but contain several examples of inferred copy
number events to enable interpretation and visualisation of the copy
number variation that is present in entire cancer genomes. Sequenza is
the tool we will use to perform this analysis. It consists of a two
Python pre-processing steps followed by a third step in R to infer the
depth ratio, cellularity, ploidy and to plot the results for
interpretation.</p>
<p>In the second part of the tutorial we will also be using IGV to
visualise and manually inspect the copy number variation we inferred in
the first part for validation purposes. This section will also include a
discussion on the importance of good quality data by highlighting the
inadequacies of the workshop dataset and the implications this has on
analysis results.</p>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>We will use a dataset derived from whole genome sequencing of a
33-yr-old lung adenocarcinoma patient, who is a never-smoker and has no
familial cancer history.</p>
<p>The data files are contained in the subdirectory called <code class="codehilite">data</code> and are
the following:</p>
<blockquote>
<p><code class="codehilite">normal.chr5.60Mb.bam</code> and <code class="codehilite">normal.chr5.60Mb.bam.bai</code></p>
<p><code class="codehilite">tumour.chr5.60Mb.bam</code> and <code class="codehilite">tumour.chr5.60Mb.bam.bai</code></p>
</blockquote>
<p><br>
These files are based on subsetting the whole genomes derived from blood
and liver metastases to the first 60Mb of chromosome 5. This will allow
our analyses to run in a sufficient time during the workshop, but it’s
worth being aware that we are analysing just 1.9% of the genome which
will highlight the length of time and resources required to perform
cancer genomics on full genomes!</p>
<p>Open the Terminal and go to the <code class="codehilite">CNV</code> working directory:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd cnv/
</pre></div>
</td></tr></table>

<div class="admonition failure">
<p>All commands entered into the terminal for this tutorial should be from
within the <strong><code class="codehilite">cnv</code></strong> directory.</p>
</div>
<p>Check that the <code class="codehilite">data</code> directory contains the above-mentioned files by
typing:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ll data
</pre></div>
</td></tr></table>

<p>All commands used in this tutorial can be copy/pasted from the
commands.sh file in the <code class="codehilite">cnv</code> directory.</p>
<hr />
<h2 id="sequenza-cnv-analysis">Sequenza CNV Analysis<a class="headerlink" href="#sequenza-cnv-analysis" title="Permanent link">&para;</a></h2>
<p>Sequenza is run in three steps. The first pre-processing step is run on
the final normal and tumour mapped data (BAM files) in order to walk the
genome in a pileup format (automatically generated by samtools). This
first step finds high quality sites in the genomes and extracts their
depth and genotype in the normal genome and calculates the variant
alleles and allele frequencies in the tumour genome. The second step is
to perform a binning on these sites to save space and analysis time in
the third step. Finally, the third step is run in R normalise the depth
ratio between the normal/tumour genomes, infer cellularity and ploidy
and graphically output results for interpretation.</p>
<h3 id="step-1-pre-processing-walking-the-genome">Step 1: Pre-Processing – Walking the Genome<a class="headerlink" href="#step-1-pre-processing-walking-the-genome" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>pypy software/sequenza/sequenza-utils.py bam2seqz -n data/normal.chr5.60Mb.bam -t data/tumour.chr5.60Mb.bam --fasta assets/human_g1k_v37.fasta -gc assets/human_g1k_v37.gc50Base.txt.gz -C 5:1-60000000 | gzip &gt; stage1.seqz.gz
</pre></div>
</td></tr></table>

<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>Press tab after typing a few characters of a directory of filename to
auto-complete the rest. This makes entering long file names very quick.</p>
</div>
<p>Explanation of parameters:</p>
<blockquote>
<p><strong>-n</strong>: the normal BAM<br />
<strong>-t</strong>: the tumour BAM<br />
<strong>--fasta</strong>: the reference genome used for mapping (b37 here)<br />
<strong>-gc</strong>: GC content as windows through the genome (pre-generated and downloadable from the Sequenza website)<br />
<strong>-C</strong>: specifies the genomic location to process  </p>
</blockquote>
<div class="admonition failure">
<p>There will not be any indication that it is running once you launch the
command, to make sure it is running open a new Terminal tab with
<code class="codehilite">Shift + Control + T</code> (or from the menu with File then Open Tab) and type
the command <code class="codehilite">top</code>. You should see the top line being the command ’pypy’ with
a % CPU usage of 98/99%. Press <code class="codehilite">q</code> to quit out of this process view and go
back to the tab running Sequenza. If everything is running correctly, it
will take approximately 40 minutes to run. Go have a coffee!</p>
</div>
<p><br>
Once the command is done you will be returned to the terminal prompt.
Make sure the output file is the correct size by typing <code class="codehilite">ll -h</code> from the
Terminal window that you ran Sequenza from, there should be a file
called <code class="codehilite">stage1.seqz.gz</code> of the size 326M.</p>
<p>You can look at the first few lines of the output in the file
<code class="codehilite">stage1.seqz.gz</code> with:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>zcat stage1.seqz.gz <span class="p">|</span> head -n <span class="m">20</span>
</pre></div>
</td></tr></table>

<p>This output has one line for each position in the BAMs and includes
information on the position, depths, allele frequencies, zygosity, GC in
the location.</p>
<h3 id="step-2-perform-binning">Step 2: Perform Binning<a class="headerlink" href="#step-2-perform-binning" title="Permanent link">&para;</a></h3>
<p>The binning step takes the rows of genomic positions and compresses them
down to 1 row for every 200 rows previously. This massively reduces the
file size and processing time in the third step.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>pypy software/sequenza/sequenza-utils.py seqz-binning -w 200 -s stage1.seqz.gz | gzip &gt; stage2.seqz.gz
</pre></div>
</td></tr></table>

<p>Explanation of parameters:</p>
<blockquote>
<p><strong>-w</strong>: the window size (typically 50 for exomes, 200 for genomes)<br />
<strong>-s</strong>: the large seqz file generated in the first step</p>
</blockquote>
<p>This step should take approximately 4 minutes to complete.</p>
<h3 id="step-3-running-sequenza-algorithms-and-plotting-results">Step 3: Running Sequenza Algorithms and Plotting Results<a class="headerlink" href="#step-3-running-sequenza-algorithms-and-plotting-results" title="Permanent link">&para;</a></h3>
<p>We will now perform the CNV analysis and output the results using the R
part of Sequenza.</p>
<p>Open the R terminal:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>R
</pre></div>
</td></tr></table>

<p>You should now see the R prompt identified with &ldquo;&gt;&rdquo;.</p>
<p>Run the Sequenza R commands:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kn">library</span><span class="p">(</span><span class="s">&quot;sequenza&quot;</span><span class="p">)</span>
<span class="kp">setwd</span><span class="p">(</span><span class="s">&quot;/home/trainee/cnv&quot;</span><span class="p">)</span>
data.file <span class="o">&lt;-</span> <span class="s">&quot;stage2.seqz.gz&quot;</span>
seqzdata <span class="o">&lt;-</span> sequenza.extract<span class="p">(</span>data.file<span class="p">)</span>
CP.example <span class="o">&lt;-</span> sequenza.fit<span class="p">(</span>seqzdata<span class="p">)</span>
sequenza.results<span class="p">(</span>sequenza.extract <span class="o">=</span> seqzdata<span class="p">,</span> cp.table <span class="o">=</span> CP.example<span class="p">,</span> sample.id <span class="o">=</span> <span class="s">&quot;CanGenWorkshop&quot;</span><span class="p">,</span> out.dir<span class="o">=</span><span class="s">&quot;sequenza_results&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>If every command ran successfully, you will now have a <code class="codehilite">sequenza_results</code>
folder containing 13 files.</p>
<p>Quit R:  </p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">q</span><span class="p">()</span>
</pre></div>
</td></tr></table>

<p>Then enter <code class="codehilite">n</code> at the &ldquo;Save workspace image&rdquo; prompt.</p>
<hr />
<h2 id="sequenza-analysis-results-and-visualisation">Sequenza Analysis Results and Visualisation<a class="headerlink" href="#sequenza-analysis-results-and-visualisation" title="Permanent link">&para;</a></h2>
<p>One of the first and most important estimates that Sequenza provides is
the tumour cellularity (the estimated percentage of tumour cells in the
tumour genome). This estimate is based on the B allele frequency and
depth ratio through the genome and is an important metric to know for
interpretation of Sequenza results and for other analyses. Lets look at
the cellularity estimate for our analysis by opening
<code class="codehilite">CanGenWorkshop_model_fit.pdf</code> with the command:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>evince sequenza_results/CanGenWorkshop_model_fit.pdf
</pre></div>
</td></tr></table>

<p><br>
The cellularity estimate is at the top along with the average ploidy
estimate and the standard deviation of the B allele frequency. We can
see that the cellularity has been estimated at 24% which is fairly low
and we will see why this is bad in the next section on CNV
visualisation. The ploidy value of 2.1 indicates this piece of the
genome is not hugely amplified or deleted and the BAF standard deviation
indicates there are no significant long losses of heterozygosity.</p>
<p>Close the PDF window to resume the Terminal prompt.</p>
<p>Let’s now look at the CNV inferences through our genomic block. Open the
genome copy number visualisation file with:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>evince sequenza_results/CanGenWorkshop_genome_view.pdf
</pre></div>
</td></tr></table>

<p><br>
This file contains three &ldquo;pages&rdquo; of copy number events through the
entire genomic block.</p>
<ol>
<li>The first page shows copy numbers of the A (red) and B (blue) alleles,</li>
<li>The second page shows overall copy number changes, and</li>
<li>The third page shows the B allele frequency and depth ratio through
genomic block.  </li>
</ol>
<p>Looking at the overall copy number changes, we see that
our block is at a copy number of 2 with a small duplication to copy
number 4 about &#8531; of the way through the block and another just after
halfway through the block. There is also a reduction in copy number to 1
copy about &#8536; of the way through the block. The gap that you see just
before this reduction in copy number is the chromosomal centromere - an
area that is notoriously difficult to sequence so always ends up in a
gap with short read data.</p>
<p>You can see how this is a very easy to read output and lets you
immediately see the frequency and severity of copy number events through
your genome. Let’s compare the small genomic block we ran with the same
output from the entire genome which has been pre-computed for you. This
is located in the <code class="codehilite">pre_generated/results_whole_genome</code> folder and
contains the same 13 output files as for the small genomic block. As
before, let’s look at the cellularity estimate with:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>evince pre_generated/results_whole_genome/CanGenWorkshop_model_fit.pdf
</pre></div>
</td></tr></table>

<p><br>
It now looks like it’s even worse at just 16%! A change is to be
expected as we were only analysing 1.9% of the genome. Let’s now look at
the whole genome copy number profile with:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>evince pre_generated/results_whole_genome/CanGenWorkshop_genome_view.pdf
</pre></div>
</td></tr></table>

<p><br>
You can see that there are a number of copy number events across the
genome and our genomic block (the first 60Mb of chromosome 5) is
inferred as mostly copy number 4 followed by a reduction to copy number
2, rather than 2 to 1 as we saw in the output we generated. The reason
for this is that Sequenza uses the genome-wide depth ratio and BAF in
order to estimate copy number, if you ask it to analyse a small block
mostly at copy number 4 with a small reduction to copy number 2, the
most likely scenario in lieu of more data is that this is a copy number
2 block with a reduction to 1. It’s important to carefully examine the
cellularity, ploidy and BAF estimates of your sample along with the
plots of model fit (<code class="codehilite">CanGenWorkshop_model_fit.pdf</code>) and
cellularity/ploidy contours (<code class="codehilite">CanGenWorkshop_CP_contours.pdf</code>) in order
to decide if you believe Sequenza’s inference of the copy numbers. Have
a look at these for yourself if you want to get a better idea of how
Sequenza makes its inferences and conclusions.</p>
<hr />
<h2 id="cnv-visualisationconfirmation-in-igv">CNV Visualisation/Confirmation in IGV<a class="headerlink" href="#cnv-visualisationconfirmation-in-igv" title="Permanent link">&para;</a></h2>
<p>Let’s see if we can visualise one of the CNV events where copy number
increased significantly. We’ll focus on the copy number 4 event seen at
about &#8531; of the way through the <code class="codehilite">CanGenWorkshop_genome_view.pdf</code> output
we generated. First, we need to find the coordinates that have been
predicted for this event. Have a look at the <code class="codehilite">CanGenWorkshop_segments.txt</code>
file in the results folder to view all predicted CNV events with:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less sequenza_results/CanGenWorkshop_segments.txt
</pre></div>
</td></tr></table>

<p><br>
There is only one at a copy number of 4 (CNt column) and it starts at
21,051,700 to 21,522,065 which is 470kb and corresponds to the small block
we see in the genome view PDF.</p>
<p>Quit out of viewing the segments file by pressing <code class="codehilite">q</code>.</p>
<p>We will now open IGV and see if we can observe the predicted increase in
copy number within these genomic coordinates.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>/home/trainee/snv/Applications/igv/igv.sh
</pre></div>
</td></tr></table>

<p>IGV will take 30 seconds or so to open so just be patient.</p>
<p><br>
For a duplication of this size, we will not be able to easily observe it
just by looking at the raw read alignments. In order to see it we will
generate two tiled data files (TDFs) within IGV which contain the
average read depth for a given window size through the genome. This
means that we can aggregate the average read depth over relatively large
chunks of the genome and compare these values between the normal and
tumour genomes.</p>
<p>To begin, we will go to <code class="codehilite">Tools</code> then <code class="codehilite">Run igvtools...</code> in the IGV
menubar. Specify the normal bam file (under <code class="codehilite">cnv</code> then <code class="codehilite">data</code>) as the
input file and change the window size to 100,000 (one hundred thousand).
Then press the <code class="codehilite">Run</code> button and IGV will make the TDF file. This takes
about 5 minutes. <strong>Repeat this for the tumour genome.</strong></p>
<p>After you have both TDF files, go to <code class="codehilite">File</code> and <code class="codehilite">Load from file...</code> in
the menubar and select the BAM and TDF files to open. Once you have
opened them, they will appear as tracks along with the BAM tracks we
loaded initially. Navigate to the genomic coordinates of our event
(5:21,051,700-21,522,065) by typing it in the coordinate box at the top.
Mouse over the two blue tracks to get the average depth values for the
100,000 bp windows. What you should see is that the liverMets sample has
3-6X more coverage than the Blood sample for the four windows that cover
this region.</p>
<p>This may seem a bit underwhelming, after all, wasn’t the increase of the
region to a copy number of 4, i.e. we expect a doubling of reads in the
tumour? To explain why we are only seeing such a small coverage
increase, we need to turn to our good friend mathematics!</p>
<p>Imagine we have two 30X genomes for the normal and tumour samples and
the tumour is at 100% purity. If there is a copy number increase to 4 in
the tumour from 2 in the normal, the duplicated segment should indeed
have twice as many reads as the same segment in the normal genome. Now,
lets imagine the tumour genome was only at a purity of 50% (i.e. it
contains 50% normal cells and 50% tumour cells). Now, half of the
duplicated &ldquo;tumour genome&rdquo; segment will be at a copy number of 2 and
half will be at 4. What does this mean when we sequence them as a
mixture? The resulting average copy number of the block will be
<script type="math/tex">(0.5*2)+(0.5*4) = 3</script>. Now what if we only have 16% tumour cells in our
&ldquo;tumour genome&rdquo;? This will be <script type="math/tex">(0.84*2)+(0.16*4) = 2.32</script>. You can see
how sequencing a low cellularity tumour at a low depth makes it much
harder to infer copy number variations!</p>
<p>Returning to our genomes at hand, when we previously looked at the
cellularity estimate of this tumour we saw it was 20% from the small
block we ran or 16% from the whole genome. Thus, the read depth increase
of just 3-6X (about 10-20% more reads) in this segment is not
surprising. A low cellularity tumour greatly reduces our power to infer
copy number events as relatively small changes in depth can occur by
chance in the genome and these can be mis-identified as copy number
changes. As well as this, it reduces our power for other analyses since
we must also remember that a tumour can itself contain multiple clones
which have to share just 16% of reads.</p>
<p>It is possible to sequence through a low-cellularity sample when, for
example, there is no way to take another sample (as is the case of most
biopsies). &ldquo;Sequencing through&rdquo; means to simply sequence the tumour at a
much higher coverage, usually 90-120X. This will mean that there will be
a net increase in reads supplying evidence for copy number events and
variants and in aggregate these will still retain power to infer these
events when using tools that look at the whole genome like Sequenza
does.</p>
<hr />
<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ol>
<li>F. Favero, T. Joshi, A. M. Marquard, N. J. Birkbak, M. Krzystanek,
    Q. Li, Z. Szallasi, and A. C. Eklund. &ldquo;Sequenza: allele-specific
    copy number and mutation profiles from tumor sequencing data&rdquo;.
    Annals of Oncology, 2015, vol. 26, issue 1, 64-70.</li>
</ol>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Single Nucleotide Variant Calling and Annotation
              </span>
            </div>
          </a>
        
        
          <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Structural Variant Analysis
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-e3caa82af6.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>