
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.1">
    
    
      
        <title>Eukaryote Assembly - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-f98ca285f9.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  DeNovo Modules
                </span>
              
            
            Eukaryote Assembly
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link md-tabs__link--active">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" checked>
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Eukaryote Assembly
      </label>
    
    <a href="./" title="Eukaryote Assembly" class="md-nav__link md-nav__link--active">
      Eukaryote Assembly
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#first-pass-genome-assembly" title="First Pass Genome Assembly" class="md-nav__link">
    First Pass Genome Assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fusarium-first-pass-with-a-goal" title="Fusarium first pass with a goal" class="md-nav__link">
    Fusarium first pass with a goal
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task11-first-pass-assembly-k71" title="Task1.1: First pass assembly, k=71" class="md-nav__link">
    Task1.1: First pass assembly, k=71
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task12-first-pass-assembly-k27" title="Task1.2: First pass assembly, k=27" class="md-nav__link">
    Task1.2: First pass assembly, k=27
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#first-pass-genome-assembly" title="First Pass Genome Assembly" class="md-nav__link">
    First Pass Genome Assembly
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fusarium-first-pass-with-a-goal" title="Fusarium first pass with a goal" class="md-nav__link">
    Fusarium first pass with a goal
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task11-first-pass-assembly-k71" title="Task1.1: First pass assembly, k=71" class="md-nav__link">
    Task1.1: First pass assembly, k=71
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#task12-first-pass-assembly-k27" title="Task1.2: First pass assembly, k=27" class="md-nav__link">
    Task1.2: First pass assembly, k=27
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#references" title="References" class="md-nav__link">
    References
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Eukaryote Assembly</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Assess general assembly approach, kmer spectra and biases.</p>
</li>
<li>
<p>Visually inspect the kmer spectra and KAT plots</p>
</li>
<li>
<p>Run a first pass eukaryotic assembly and do goal checks</p>
</li>
<li>
<p>Develop validation metrics or tools for NGS data and assembly.</p>
</li>
<li>
<p>Improving methods and pipelines for genome assembly.</p>
</li>
<li>
<p>Convince the lab guys to tweak protocols.</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Kmer Analysis Tool kit:<br />
<a href="https://github.com/TGAC/KAT">https://github.com/TGAC/KAT</a></p>
<p>Nextclip:<br />
<a href="https://github.com/richardmleggett/nextclip">https://github.com/richardmleggett/nextclip</a></p>
<p>Abyss:<br />
<a href="http://www.bcgsc.ca/platform/bioinfo/software/abyss">http://www.bcgsc.ca/platform/bioinfo/software/abyss</a></p>
<p>Soap Denovo:<br />
<a href="http://soap.genomics.org.cn/soapdenovo.html">http://soap.genomics.org.cn/soapdenovo.html</a></p>
<p>SOAPec:<br />
<a href="http://soap.genomics.org.cn/about.html">http://soap.genomics.org.cn/about.html</a></p>
<p>BLAST:<br />
<a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi">http://blast.ncbi.nlm.nih.gov/Blast.cgi</a></p>
<h2 id="first-pass-genome-assembly">First Pass Genome Assembly<a class="headerlink" href="#first-pass-genome-assembly" title="Permanent link">&para;</a></h2>
<p>Assuming by now you are familiar with the general concept of <em>de novo</em>
assembly, kmers and the de Bruijn graph based assembler. In this
tutorial we will use ABySS to perform the first pass assembly of a
eukaryotic genome and look at various parameters to assess the
information content of the input data and choice of assembly parameters.
sequence data.\
Genome assembly is a challenging problem requiring heavy computational
resources, expertise and time. Before you beging the process of denovo
assembly there are a number of points you need to consider:</p>
<p>What is the objective of your assembly experiment? What biological
question(s) you have?</p>
<p>Is assembly strictly neccessary for the purpose in question?</p>
<p>Do you have right kind of data and enough coverage to start with?</p>
<p>Do you have suitable computaitonal resources to run this assembly?</p>
<p>Remember that the assembly is just a probabilistic model of a genome,
condensing the information from the experimental evidence. All the
information is already present in the experimental results. The goal of
the assembly is to find the right motifs, the correct number of times,
in correct order and position.</p>
<h3 id="fusarium-first-pass-with-a-goal">Fusarium first pass with a goal<a class="headerlink" href="#fusarium-first-pass-with-a-goal" title="Permanent link">&para;</a></h3>
<p>Goal: Identify a fusarium sample is ``closer&rdquo; to <em>F. graminearum</em> or
<em>F.pseudograminearum</em></p>
<ul>
<li>
<p>Previous knowledge</p>
<ul>
<li><em>F. graminearum</em> has a cluster producing PKS6 and NRSP7, while
    <em>F. pseudograminearum</em> produces PKS40 and NRPS32</li>
</ul>
</li>
<li>
<p>Data</p>
<ul>
<li>
<p>Proteins sequences for:</p>
<ul>
<li>
<p><em>F. graminearum</em> (non necrotrophic): PKS6 and NRSP7</p>
</li>
<li>
<p><em>F. pseudograminearum</em> (necrotrophic): PKS40 and NRPS32</p>
</li>
</ul>
</li>
</ul>
</li>
<li>
<p>Strategy:</p>
<ul>
<li>Check PKS6-NRSP7 and PKS40-NRPS32 cluster presence.</li>
</ul>
</li>
<li>
<p>Assembly goals:</p>
<ul>
<li>
<p>Assembly goal (I): to capture a good enough representation of
    the protein-coding space to get blast matches</p>
</li>
<li>
<p>Assembly goal (II): to accurately represent the relevant whole
    cluster loci in a single sequence.</p>
</li>
</ul>
</li>
</ul>
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>Open the Terminal. First, go to the right folder, where the data are
stored.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd /home/trainee/eukaryotic
ls
</pre></div>
</td></tr></table>

<h2 id="task11-first-pass-assembly-k71">Task1.1: First pass assembly, k=71<a class="headerlink" href="#task11-first-pass-assembly-k71" title="Permanent link">&para;</a></h2>
<p>Let’s assemble <em>Fusarium</em> with abyss, k=71</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd /home/trainee/eukaryotic
mkdir abyss_k71
cd abyss_k71
abyss-pe in=&quot;../CS3270_A8733_GCCAAT_L001_R1.fastq ../CS3270_A8733_GCCAAT_L001_R2.fastq&quot; k=71 name=CS3270_abyss_k71 np=4 &gt; CS3270_abyss_k71.log 2&gt;&amp;1
</pre></div>
</td></tr></table>

<p>Description of the arguments used in the command:</p>
<p>k
:   = kmer size</p>
<p>np
:   = number of processors to be used</p>
<p>sequence file names
:   = R1 and R2 reads of a paired end sequence data</p>
<p>Let’s look at the statistics of the assembly we just did&hellip;</p>
<p>Ok, there is no stats available in the folder, but we can always use
<code class="codehilite">abyss-fac</code> to get the stats:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>abyss-fac CS3270_abyss_k71-*tigs.fa  | tee CS3270_abyss_k71-stats.tab
less CS3270_abyss_k71-stats.tab
</pre></div>
</td></tr></table>

<p>[H]</p>
<p>lllllllllll</p>
<p><strong>n</strong> &amp; <strong>n:500</strong> &amp; <strong>L50</strong> &amp; <strong>min</strong> &amp; <strong>N80</strong>&amp; <strong>N50</strong>&amp; <strong>N20</strong>&amp;
<strong>E-size</strong>&amp; <strong>max</strong> &amp; <strong>sum</strong>&amp; <strong>name</strong>\
27 &amp; 13 &amp; 2 &amp; 970 &amp; 6004 &amp; 13202 &amp; 52602 &amp; 28712 &amp; 52602 &amp; 112849 &amp;
CS3270_abyss_k71-unitigs.fa%\
5 &amp; 1 &amp; 1 &amp; 128429 &amp; 128429 &amp; 128429 &amp; 128429 &amp; 128429 &amp; 128429 &amp; 128429
&amp; CS3270_abyss_k71-contigs.fa%\</p>
<p>[tab:fusariumk71]</p>
<p>How many unitigs/contigs do you have in the assembly?\</p>
<p>27/5\</p>
<p>What are the length statistics of your assembly?\</p>
<p>in the table above\</p>
<p>Does it match what you think before the assembly and why?\</p>
<p>No\</p>
<p>The assembly is looking strange! It’s time for some analysis:</p>
<p>Check frequencies for kmers kept/discarded/etc.</p>
<p>Check spectra-cn and compare with expectations. Let’s do this by the
following commands:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less CS3270_abyss_k71.log
less coverage.hist
</pre></div>
</td></tr></table>

<p>We will now plot the values from the <code class="codehilite">coverage.hist</code>:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gnuplot &lt;Press enter&gt;
gnuplot&gt; set xrange [0:200]
gnuplot&gt; set yrange [0:5000]
gnuplot&gt; plot &quot;coverage.hist&quot;
Type exit or quit to leave the gnuplot
gnuplot&gt; exit
</pre></div>
</td></tr></table>

<p>Looks like we are not assembling this bit, let’s have another look at
the spectra</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>kat comp -o reads_vs_abyss_k71 -t 4 -C -D --d1_bins 2000 &#39;../*.fastq&#39; CS3270_abyss_k71-contigs.fa
</pre></div>
</td></tr></table>

<p>Description of the arguments used in the command:</p>
<p>o
:   = Path prefix for files generated by this program.</p>
<p>t
:   = The number of threads to use.</p>
<p>C
:   = Whether the jellyfish hash for input 1 contains K-mers produced
    for both strands</p>
<p>D
:   = Whether the jellyfish hash for input 2 contains K-mers produced
    for both strands</p>
<p>–d1_bins
:   = Number of bins for the first dataset. i.e. number of rows in the
    matrix</p>
<!-- -->

<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>kat plot spectra-cn -y 600 -x 2000 -o reads_vs_abyss1-main.mx.spectra-cn_2000.png reads_vs_abyss_k71-main.mx
</pre></div>
</td></tr></table>

<p>Description of the arguments used in the command:</p>
<p>x
:   = Maximum value for the x-axis (default value auto calculated from
    matrix, otherwise 1000)</p>
<p>y
:   = Maximum value for the y-axis (default value auto calculated from
    matrix if possible, otherwise, 1000000)</p>
<p>o
:   = The path to the output file</p>
<p>The kmer spectra for <em>Fusarium</em> assembly with abyss, k=71 should be
looking like this:</p>
<p>[H] <img alt="image" src="../images/abyss-k71.png" /> [fig:fusariumk71]</p>
<p>Take the output and BLAST it in NCBI. What is it? Surprising?</p>
<p>Choosing a wrong k value (too large in this case) and just running a
typical assembly job, we can end up with something quite more
interesting. It is easy by comparison to spot some missing content,
alongside duplications and triplications (and quadruplications and so
on) that should not be there. This assembly will get us nowhere, let’s
choose a lower K to gain coverage and start again.</p>
<h2 id="task12-first-pass-assembly-k27">Task1.2: First pass assembly, k=27<a class="headerlink" href="#task12-first-pass-assembly-k27" title="Permanent link">&para;</a></h2>
<p>We now assemble <em>fusarium</em> with <code class="codehilite">abyss</code> and k=27:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd /home/trainee/eukaryotic
mkdir abyss_k27
cd abyss_k27
abyss-pe in=&quot;../CS3270_A8733_GCCAAT_L001_R1.fastq ../CS3270_A8733_GCCAAT_L001_R2.fastq&quot; k=27 name=CS3270_abyss_k27 np=4 &gt; CS3270_abyss_k27.log 2&gt;&amp;1
</pre></div>
</td></tr></table>

<p>Let’s look at the stats by doing:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less CS3270_abyss_k27-stats.tab
</pre></div>
</td></tr></table>

<p>Stats look better:</p>
<p>[H]</p>
<p>lllllllllll</p>
<p><strong>n</strong> &amp; <strong>n:500</strong> &amp; <strong>L50</strong> &amp; <strong>min</strong> &amp; <strong>N80</strong>&amp; <strong>N50</strong>&amp; <strong>N20</strong>&amp;
<strong>E-size</strong>&amp; <strong>max</strong> &amp; <strong>sum</strong>&amp; <strong>name</strong>\
30645 &amp; 2717 &amp; 430 &amp; 502 &amp; 11354 &amp; 25336 &amp; 47966 &amp; 31027 &amp; 147694 &amp;
36.14e6 &amp; CS3270_abyss_k27-unitigs.fa\
21511 &amp; 350 &amp; 33 &amp; 527 &amp; 157565 &amp; 338989 &amp; 630228 &amp; 407098 &amp; 1265237 &amp;
36.52e6 &amp; CS3270_abyss_k27-contigs.fa\
21327 &amp; 205 &amp; 17 &amp; 527 &amp; 332444 &amp; 716132 &amp; 1265237 &amp; 791882 &amp; 1880850 &amp;
36.51e6 &amp; CS3270_abyss_k27-scaffolds.fa\</p>
<p>[tab:fusariumk27]</p>
<p>Let’s check a bit anyway:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less CS3270_abyss_k27.log
less coverage.hist
</pre></div>
</td></tr></table>

<p>How is the coverage plot looking now?</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gnuplot
gnuplot&gt; set xrange [0:50]
gnuplot&gt; set xrange [0:4000000]
gnuplot&gt; plot &quot;coverage.hist&quot;
gnuplot&gt; exit
</pre></div>
</td></tr></table>

<p>K-mer spectrum:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>kat plot spectra-cn -y 1000 -x 1000 -o reads_vs_abyss1-main.mx.spectra-cn_noabsent.png reads_vs_abyss1-main.mx
kat comp -o reads_vs_abyss_k27 -t 4 -C -D &#39;../*.fastq&#39; CS3270_abyss_k27-scaffolds.fa
</pre></div>
</td></tr></table>

<p>[H] <img alt="image" src="../images/abyss-k27.png" /> [fig:fusariumk27]</p>
<p>Any tools you can use to check kmer spectra at any K before assembling?\</p>
<p>KAT\</p>
<p>Can you predict what will happen if you use KAT with larger K values?\</p>
<p>Will the assembly answer the biological question?</p>
<p>Use BLAST or BLAT and the databases to check&hellip;</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>makeblastdb -in CS3270_abyss_k27-scaffolds.fa -dbtype nucl
blat –t=dnax –q=prot –minIdentity=90 CS3270_abyss_k27-scaffolds.fa test_genes.fasta out.psl
</pre></div>
</td></tr></table>

<h2 id="references">References<a class="headerlink" href="#references" title="Permanent link">&para;</a></h2>
<ol>
<li>De novo genome assembly: what every biologist should know Nature
    Methods 9, 333 – 337 (2012) doi:10.1038/nmeth.1935</li>
</ol>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Bacterial Assembly
              </span>
            </div>
          </a>
        
        
          <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Scaffolding
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-e3caa82af6.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>