
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.1">
    
    
      
        <title>Metagenomics Visualisation - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-f98ca285f9.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Metagenomics Modules
                </span>
              
            
            Metagenomics Visualisation
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../cancer-module-alignment/alignment/" title="Cancer Genomics Modules" class="md-tabs__link">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link md-tabs__link--active">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-alignment/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7" checked>
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Metagenomics Visualisation
      </label>
    
    <a href="./" title="Metagenomics Visualisation" class="md-nav__link md-nav__link--active">
      Metagenomics Visualisation
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-sets" title="Data sets" class="md-nav__link">
    Data sets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#beta-diversity-computation-and-plots" title="Beta Diversity computation and plots" class="md-nav__link">
    Beta Diversity computation and plots
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the environment" class="md-nav__link">
    Prepare the environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emperor" title="Emperor" class="md-nav__link">
    Emperor
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data-sets" title="Data sets" class="md-nav__link">
    Data sets
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#beta-diversity-computation-and-plots" title="Beta Diversity computation and plots" class="md-nav__link">
    Beta Diversity computation and plots
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the environment" class="md-nav__link">
    Prepare the environment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emperor" title="Emperor" class="md-nav__link">
    Emperor
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="analysis-visualisation">Analysis Visualisation<a class="headerlink" href="#analysis-visualisation" title="Permanent link">&para;</a></h1>
<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<hr />
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Visualise between sample comparisons, using Emperor PCA</p>
</li>
<li>
<p>Understand the difference between weighted and unweighted analysis</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<hr />
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Emperor: <br />
<a href="http://qiime.org/emperor/tutorial_index.html">http://qiime.org/emperor/tutorial_index.html</a></p>
<h3 id="data-sets">Data sets<a class="headerlink" href="#data-sets" title="Permanent link">&para;</a></h3>
<hr />
<p>Sutton et al. (2013). Impact of Long-Term Diesel Contamination on Soil Microbial Community Structure: <br />
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3553749/pdf/zam619.pdf">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3553749/pdf/zam619.pdf</a></p>
<p>Fierer et al. (2010). Forensic identification using skin bacterial communities: <br />
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2852011/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2852011/</a></p>
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<hr />
<p>Good data quality and sample metadata is important for visualising
metagenomics analysis.</p>
<p>For this tutorial we are using Emperor a browser enabled scatter plot
visualisation tool. We will be using the Sutton and Fierer data sets for
viewing the principal components analysis (PCoA) from the QIIME Beta
diversity analysis (16S).</p>
<h2 id="beta-diversity-computation-and-plots">Beta Diversity computation and plots<a class="headerlink" href="#beta-diversity-computation-and-plots" title="Permanent link">&para;</a></h2>
<hr />
<p>Beta diversity represents between sample comparisons based on their
composition. The output of these comparisons is a square matrix where a
“distance” or dissimilarity is calculated between every pair of
community samples, reflecting the dissimilarity between those samples.
The data distance matrix can be then visualized with analyses such as
PCoA and hierarchical clustering. PCoA is a technique that maps the
samples in the distance matrix to a set of axes that show the maximum
amount of variation explained. The principal coordinates can be plotted
in two or three dimensions to provide an intuitive visualization of the
data structure to look at differences between the samples, and look for
similarities by sample conditions.</p>
<p>The Beta diversity workflow you ran earlier contained a number of steps:</p>
<ul>
<li>
<p>Rarifying the OTU table to remove sample heterogeneity. A rarified
    OTU table should be used so that artificial diversity induced due to
    different sampling effort is removed.</p>
</li>
<li>
<p>Calculating the Beta diversity, the unifrac weighted and unweighted
    generated principal coorodinates are created here by default.</p>
</li>
</ul>
<p>The following are the required options and inputs for Beta diversity
analysis:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>1.  Option -i, OTU table (*.biom)
2.  Option –t, phylogeny tree (*.tre) from pick_de_novo_otus.py
3.  Option –m, the user-defined mapping file
4.  Option –o, the output directory
5.  Option –e, the number of sequences per sample (sequencing depth)
</pre></div>
</td></tr></table>

<p>You do not need to run the following command, it may overwrite the
pre-computed analysis. This is the command you ran on the Sutton sub
dataset.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>beta_diversity_through_plots.py -i otus/otu_table.biom -m mapping.txt -o wf_bdiv_even122/ -t otus/rep_set.tre -e 122
</pre></div>
</td></tr></table>

<h3 id="prepare-the-environment">Prepare the environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h3>
<p>The data for this practical can be found in the Taxonomy directory on
your desktop. Go to the pre-computed Beta diversity analysis in the
Taxonomy directory.</p>
<p>Open the Terminal and go to where the data is stored. Investigate the
directories available.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd ~/Desktop/Taxonomy/sutton_full_denoised/
ls –lhtr
</pre></div>
</td></tr></table>

<p>Open the following files: wf_bdiv_even394/unweighted_unifrac_pc.txt
and wf_bdiv_even394/weighted_unifrac_pc.txt using the ‘less’
command. The option –S with ‘less’ allows you to view files with
unwrapped lines, to escape type q (for quit). To scroll left and right,
use the arrow keys.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>less –S wf_bdiv_even394/unweighted_unifrac_pc.txt
less –S wf_bdiv_even394/weighted_unifrac_pc.txt
</pre></div>
</td></tr></table>

<p>In each file every sample is listed in the first column, and the
subsequent columns contain the value for the sample against the noted
principal coordinate. At the bottom of each Principal Coordinate column,
you will find the eigen value and percent of variation explained by the
coordinate. Note the major difference between weighted and unweighted
analysis is the inclusion of OTU abundance when calculating distances
between communities. You should use weighted if the biological question
you are trying to ask takes OTU abundance of your groups into
consideration. If some samples are forming groups with weighted, then
likely the larger or smaller abundances of several OTUs are the primary
driving force in PCoA space, but when all OTUs are considered at equal
abundance, these differences are lost (unweighted).</p>
<p>Note discrete vs. continuous only has to do with coloring of the points,
and not the calculation of UniFrac distances.</p>
<h3 id="emperor">Emperor<a class="headerlink" href="#emperor" title="Permanent link">&para;</a></h3>
<p>We can now run Emperor to prepare the PCoA plots for visualisation.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>make_emperor.py -i wf_bdiv_even394/unweighted_unifrac_pc.txt -m sutton_mapping_file.txt –b &quot;Source,Condition,Source&amp;&amp;Condition&quot; –o emporer_output_unweighted

ls –l emporer_output_unweighted
</pre></div>
</td></tr></table>

<p>Run the last make_emporer.py command with the weighted_unifrac, using
the same command. Make sure you rename the output directory to
“emporer_output_weighted“. Look in the Emporer output directories and
note the index.html file outputs.</p>
<p>To view the index.html files created from the ‘make_emperor.py’ steps
above a browser like Firefox or Chrome can be used.</p>
<p>Important - To view the Emporer plots we need to switch from the VM to
your local machine. Open the Firefox browser and view the pre-computed
Emporer directories available at the following URL.
<a href="http://www.ebi.ac.uk/~sterk/emperor/">http://www.ebi.ac.uk/~sterk/emperor/</a></p>
<p>View the unweighted analysis on the Sutton sub data set:</p>
<p><a href="http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_unweighted/">http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_unweighted/</a></p>
<p>Check the ‘Colors’ options to view your samples. What can you determine
from the PCoA plot?</p>
<p>To change the colouring scheme click the ‘Colors’ tab. The available
metadata categories are sorted in alphabetical order. Extra information
on the features can be found at
<a href="http://qiime.org/emperor/description_index.html">http://qiime.org/emperor/description_index.html</a> Try changing the
scaling, visibility, label etc. to improve the display. Note that
Emperor generates publication quality figures in scalable vector
graphics (SVG) format.</p>
<p>Now look at the weighted unifrac analysis,
<a href="http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_weighted/">http://www.ebi.ac.uk/~sterk/emperor/sutton_subset_emporer_output_weighted/</a></p>
<p>You can compare the sub data analysis against the full-denoised dataset.
<ttp://www.ebi.ac.uk/~sterk/emperor/sutton_full_denoised/></p>
<p>What are the main differences between the weighted and unweighted plots?</p>
<p>Did you notice any differences between the analysis on the sub data set
and the full denoised?</p>
<p>What are your final conclusions on the Sutton dataset?</p>
<p>What did each of the visualisation methods describe?</p>
<p>Taxonomy summary plots:</p>
<p>Alpha diversity plots:</p>
<p>Beta diversity PCoA plots:</p>
<p>Now that you are familiar with Emperor we will next view the forensic
analysis of the Fierer et. al. data. This experiment matches the
bacteria on surfaces to the skin-associated bacteria of the individual
who touched the surface. This study shows that skin-associated bacteria
can be readily recovered from surfaces (including single computer keys
and computer mice) and that the structure of these communities can be
used to differentiate objects handled by different individuals, even if
those objects have been left untouched for up to 2 weeks at room
temperature.</p>
<p>Please look the information contained in the mapping file and identify
those fields that can be plotted.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd ~/Desktop/Taxonomy/fierer/
less –S fierer_mapping_file.txt
</pre></div>
</td></tr></table>

<p>List any fields that can be plotted for this study. What is the
remainder of the information for?</p>
<p>Run Emperor to prepare the PCoA plots for visualisation</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>make_emperor.py -i fierer_unweighted_unifrac_pc.txt -m fierer_mapping_file.txt -b \ &quot;HOST_SUBJECT_ID,ENV_FEATURE,HOST_SUBJECT_ID&amp;&amp;ENV_FEATURE” -o emporer_output_unweighted
</pre></div>
</td></tr></table>

<p>Go to
<a href="http://www.ebi.ac.uk/~sterk/emperor/fierer_emporer_output_unweighted/">http://www.ebi.ac.uk/~sterk/emperor/fierer_emporer_output_unweighted/</a></p>
<p>What did you observe about the clustering?</p>
<p>What fields have been selected?</p>
<p>The on-line Qiime tutorials are very good and growing in numbers. It is
highly recommended checking out in future the many analysis options.</p>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metagenomics Taxonomic
              </span>
            </div>
          </a>
        
        
          <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Metagenomics WGA
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-e3caa82af6.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>