
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.1">
    
    
      
        <title>Read Alignment - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-f98ca285f9.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-8817cfa535.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-logo md-header-nav__button">
            <img src="../../../media/logos/bpalogo.png" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Cancer Genomics Modules
                </span>
              
            
            Read Alignment
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="md-header-nav__source">
          
        </div>
      </div>
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="BTP Modules" class="md-tabs__link">
          BTP Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="./" title="Cancer Genomics Modules" class="md-tabs__link md-tabs__link--active">
          Cancer Genomics Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../denovo-module-cli/commandline/" title="DeNovo Modules" class="md-tabs__link">
          DeNovo Modules
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics Modules" class="md-tabs__link">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-logo md-nav__button">
        <img src="../../../media/logos/bpalogo.png">
      </i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_cancer/" title="Timetable" class="md-nav__link">
      Timetable
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      BTP Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        BTP Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-chip-seq/chip-seq-draft-SL_ST/" title="ChIP-seq" class="md-nav__link">
      ChIP-seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-denovo-canu/denovo_canu/" title="DeNovo Canu" class="md-nav__link">
      DeNovo Canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-qc/ngs-qc/" title="Data Quality" class="md-nav__link">
      Data Quality
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-mapping/alignment/" title="Read Alignment" class="md-nav__link">
      Read Alignment
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-rna-seq/rna-seq/" title="RNA-Seq" class="md-nav__link">
      RNA-Seq
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-velvet/velvet/" title="de novo Genome Assembly" class="md-nav__link">
      de novo Genome Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-nav__link">
      NGS Biological Insight
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Cancer Genomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Cancer Genomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Read Alignment
      </label>
    
    <a href="./" title="Read Alignment" class="md-nav__link md-nav__link--active">
      Read Alignment
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#alignment" title="Alignment" class="md-nav__link">
    Alignment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#manipulating-sam-output" title="Manipulating SAM output" class="md-nav__link">
    Manipulating SAM output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#post-alignment-visualisation-option" title="Post Alignment Visualisation option" class="md-nav__link">
    Post Alignment Visualisation option
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-nav__link">
      Single Nucleotide Variant Calling and Annotation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-cnv/cnv-tut/" title="Copy Number Variation" class="md-nav__link">
      Copy Number Variation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-sv/sv_tut/" title="Structural Variant Analysis" class="md-nav__link">
      Structural Variant Analysis
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-viz/visu/" title="Variant Visualisation" class="md-nav__link">
      Variant Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-6" type="checkbox" id="nav-5-6">
    
    <label class="md-nav__link" for="nav-5-6">
      Somatic Mutations
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-6">
        Somatic Mutations
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/01_signatures/" title="Assessing somatic mutational signatures" class="md-nav__link">
      Assessing somatic mutational signatures
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../cancer-module-somatic/02_intogen/" title="Cohort analysis for the identification of driver genes" class="md-nav__link">
      Cohort analysis for the identification of driver genes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      DeNovo Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        DeNovo Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-cli/commandline/" title="DeNovo Command Line" class="md-nav__link">
      DeNovo Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-bac/bac_asm/" title="Bacterial Assembly" class="md-nav__link">
      Bacterial Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-euk/eukaryotic_assem/" title="Eukaryote Assembly" class="md-nav__link">
      Eukaryote Assembly
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../denovo-module-scaffolding/scaffold/" title="Scaffolding" class="md-nav__link">
      Scaffolding
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-vis/vis/" title="Metagenomics Visualisation" class="md-nav__link">
      Metagenomics Visualisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#sources-of-data" title="Sources of Data" class="md-nav__link">
    Sources of Data
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prepare-the-environment" title="Prepare the Environment" class="md-nav__link">
    Prepare the Environment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#alignment" title="Alignment" class="md-nav__link">
    Alignment
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#manipulating-sam-output" title="Manipulating SAM output" class="md-nav__link">
    Manipulating SAM output
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#post-alignment-visualisation-option" title="Post Alignment Visualisation option" class="md-nav__link">
    Post Alignment Visualisation option
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Read Alignment</h1>
                
                <h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<p>After completing this practical the trainee should be able to:</p>
<ul>
<li>
<p>Perform the simple NGS data alignment task against reference data.</p>
</li>
<li>
<p>Learn about the SAM/BAM formats for further manipulation.</p>
</li>
<li>
<p>Be able to sort and index BAM format for visualisation purposes.</p>
</li>
</ul>
<hr />
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>BWA Burrows-Wheeler Algorithm:<br />
<a href="http://bio-bwa.sourceforge.net">http://bio-bwa.sourceforge.net</a></p>
<p>Samtools:<br />
<a href="http://picard.sourceforge.net/">http://picard.sourceforge.net/</a></p>
<h3 id="useful-links">Useful Links<a class="headerlink" href="#useful-links" title="Permanent link">&para;</a></h3>
<p>SAM Specification:<br />
<a href="http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a></p>
<p>Explain SAM Flags:<br />
<a href="http://picard.sourceforge.net/explain-flags.html">http://picard.sourceforge.net/explain-flags.html</a></p>
<h3 id="sources-of-data">Sources of Data<a class="headerlink" href="#sources-of-data" title="Permanent link">&para;</a></h3>
<p><a href="http://sra.dnanexus.com/studies/ERP001071">http://sra.dnanexus.com/studies/ERP001071</a></p>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>The goal of this hands-on session is to perform an NGS alignment on the
sequencing data coming from a tumour and normal group of samples. We
will align raw sequencing data to the human genome using the BWA aligner
and then we will discuss the sequence alignment and mapping format
(SAM). SAM to BAM conversion, indexing and sorting will also be
demonstrated. These are important and essential steps for downstream
processing of the aligned BAM files.</p>
<p>This data is the whole genome sequencing of a lung adenocarcinoma
patient AK55. It was <a href="http://sra.dnanexus.com/studies/ERP001071">downloaded</a> from
<code class="codehilite">ERP001071</code>. Only the HiSeq2000 data for <code class="codehilite">Blood</code> and <code class="codehilite">liverMets</code> were analysed.</p>
<p>Accession numbers associated with read data assigned by the European
Bioinformatics Institute (EBI) start with ’ER’. e.g. ERP is the study ID
and ERR is the run. The original FASTQ files downloaded had the ERR
number in front of each read name in the FASTQ file. The read name had
to be edited to remove the ERR number at the start of the name. This had
caused problems for downstream programs such as Picard for marking
optical duplicates.</p>
<p>We have used 4 Blood samples (8 PE <code class="codehilite"><span class="na">.fastq.gz</span></code> files) and 5 Liver
samples (10 PE <code class="codehilite"><span class="na">.fastq.gz</span></code> files) data from this study to perform the
whole genome alignment using the BWA aligner. The whole process took
>150K CPU seconds per sample and the precomputed alignment will be used
in different sections of this workshop.</p>
<hr />
<h2 id="prepare-the-environment">Prepare the Environment<a class="headerlink" href="#prepare-the-environment" title="Permanent link">&para;</a></h2>
<p>By now you know about the raw sequence fastq format generated by the
Illumina sequencers. Next we will see how fastq files are aligned with
the reference genome and what are the resulting standard alignment
format. In the interest of time we have selected only 1 million paired
reads from a <code class="codehilite">Blood</code> sample to demonstrate a BWA command. The remaining
alignments have already been performed for you and will be required in
the subsequent modules of the workshop.</p>
<p>The input data for this section can be found in the <code class="codehilite">alignment</code>
directory on your desktop. Please follow the commands below to go to the
right folder and view top 10 lines of the input fastq file:</p>
<p>Open the Terminal.</p>
<p>First, go to the right folder, where the data are stored.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd /home/trainee/alignment
ls
zless input/SM_Blood_ID_ERR059356.subset_R1.fastq.gz
</pre></div>
</td></tr></table>

<hr />
<h2 id="alignment">Alignment<a class="headerlink" href="#alignment" title="Permanent link">&para;</a></h2>
<p>You already know that there are a number of competing tools for short
read alignment, each with its own set of strengths, weaknesses, and
caveats. Here we will use BWA, a widely used aligner based on the
Burrows-Wheeler Algorithm. The alignment involves two steps:  </p>
<p>1) Indexing the genome.<br />
2) Running the alignment command.  </p>
<p>BWA is a software package for mapping low-divergent sequences against a large
reference genome, such as the human genome. It consists of three
algorithms: BWA-backtrack, BWA-SW and BWA-MEM. The first algorithm is
designed for Illumina sequence reads up to 100bp, while the rest two for
longer sequences ranged from 70bp to 1Mbp. BWA-MEM and BWA-SW share
similar features such as long-read support and split alignment, but
BWA-MEM, which is the latest, is generally recommended for high-quality
queries as it is faster and more accurate. BWA-MEM also has better
performance than BWA-backtrack for 70-100bp Illumina reads. For more details see
the <a href="http://bio-bwa.sourceforge.net/bwa.shtml">BWA manual</a>.</p>
<p>BWA has a number of parameters in order to perform the alignment. To
view them all type</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa &lt;press enter&gt;
</pre></div>
</td></tr></table>

<p><br>
BWA uses an indexed genome for the alignment in order to keep its memory
footprint small. Because of time constraints we will not be running the
indexing command. It is run only once for a version of genome, and the
complete command to index the human genome version hg19 is given below.</p>
<div class="admonition failure">
<p class="admonition-title">STOP</p>
<p>You <strong>DO NOT</strong> need to run this command. This has already been run for you.</p>
<p><code class="codehilite">bwa index -p bwaIndex/human_g1k_v37.fasta -a bwtsw human_g1k_v37.fasta</code></p>
</div>
<p>We have used the following arguments for the indexing of the genome.</p>
<blockquote>
<p><strong>-p</strong>:  Prefix of the output database [same as db filename].  <br />
<strong>-a</strong>:  Algorithm for constructing BWT index. This method works with the
    whole human genome.<br />
<strong>Ref genome filename</strong>: the last argument is the name of the reference genome
file in the fasta format.</p>
</blockquote>
<p>This command will output 6 files that constitute the index. These files
that have the prefix <code class="codehilite">human_v37</code> are stored in the <code class="codehilite">bwa_index</code>
subdirectory. To view if they files have been successfully created type:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ls -l bwaIndex
</pre></div>
</td></tr></table>

<p><br>
Now that the genome is indexed we can move on to the actual alignment.
The first argument for <code class="codehilite">bwa</code> is the basename of the index for the genome
to be searched. In our case this is <code class="codehilite">human_g1k_v37.fasta</code>.</p>
<p>Align the reads from <code class="codehilite">Blood</code> samples using the following command:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>bwa mem -M -t 4 -R @RG\tSM:Blood\tID:ERR059354\tLB:lb\tPL:ILLUMINA human_g1k_v37 SM_Blood_ID_ERR059356.subset_R1.fastq.gz SM_Blood_ID_ERR059356.subset_R2.fastq.gz &gt; SM_Blood.sam
</pre></div>
</td></tr></table>

<p>The above command outputs the alignment in SAM format and stores them in
the file <code class="codehilite">SM_Blood.sam</code>.</p>
<p>We have used the following arguments for the alignment of the reads.</p>
<blockquote>
<p><strong>mem</strong>: fast mode of high quality input such the Illumina<br />
<strong>-M</strong>: flags extra hits as secondary. This is needed for compatibility with
  other tools downstream.<br />
<strong>-t</strong>: Number of threads.<br />
<strong>-R</strong>: Complete read group header line.</p>
</blockquote>
<p><br>
The <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM (Sequence Alignment/Map)</a>
format is currently the de facto standard for storing large nucleotide
sequence alignments. It is a TAB-delimited text format consisting of a header section, which is
optional, and an alignment section. If present, the header must be prior
to the alignments. Header lines start with <code class="codehilite">@</code>, while alignment lines do
not. Each alignment line has 11 mandatory fields for essential alignment
information such as mapping position.</p>
<p>Look at the top 10 lines of the SAM file by typing:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>head -n 10 SM_Blood.sam
</pre></div>
</td></tr></table>

<p><br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>Can you distinguish between the header of the SAM format and the actual
alignments?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>The header line starts with the letter ‘@’ i.e.<br />
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>@HD  VN:1.0       SO:unsorted             
@SQ  SN:chr1      LN:195471971            
@PG  ID:Bowtie2   PN:bowtie2    VN:2.2.4   CL:“/tools/bowtie2/bowtie2-default/bowtie2-align-s–wrapper basic-0 -x bowtie\_index/mm10 -q Oct4.fastq”  
</pre></div>
</td></tr></table></p><p>The actual alignments start with read ID i.e.<br />
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>SRR002012.45   0    etc    
SRR002012.48   16   chr1  
</pre></div>
</td></tr></table></p></details></div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What kind of information does the header provide?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><ul><li><p>@HD: Header line; VN: Format version; SO: the sort order of
    alignments.</p></li><li><p>@SQ: Reference sequence information; SN: reference sequence name;
    LN: reference sequence length.</p></li><li><p>@PG: Program; ID: Program record identifier; VN: Program
    version; CL: the command line that produces the alignment.</p></li></ul></details></div>
</div>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>To which chromosome are the reads mapped?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Chromosome 1.</p></details></div>
</div>
<hr />
<h2 id="manipulating-sam-output">Manipulating SAM output<a class="headerlink" href="#manipulating-sam-output" title="Permanent link">&para;</a></h2>
<p>SAM files are rather big and when dealing with a high volume of NGS
data, storage space can become an issue. As we have already seen, we can
convert SAM to BAM files (their binary equivalent that are not human
readable) that occupy much less space.</p>
<p>Convert SAM to BAM using <code class="codehilite">samtools view</code> and store the output in the
file <code class="codehilite">SM_Blood.bam</code>. You have to instruct <code class="codehilite">samtools view</code> that the input
is in SAM format (<code class="codehilite">-S</code>), the output should be in BAM format (<code class="codehilite">-b</code>) and
that you want the output to be stored in the file specified by the <code class="codehilite">-o</code>
option:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools view -bSo SM_Blood.bam SM_Blood.sam
</pre></div>
</td></tr></table>

<p><br></p>
<div class="admonition note">
<p class="admonition-title"><em>Advanced exercise</em><p>Compute summary stats for the Flag values associated with the alignments
 using:
 <table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools flagstat SM_Blood.bam
</pre></div>
</td></tr></table></p>
</p>
</div>
<hr />
<h2 id="post-alignment-visualisation-option">Post Alignment Visualisation option<a class="headerlink" href="#post-alignment-visualisation-option" title="Permanent link">&para;</a></h2>
<p>IGV is a stand-alone genome browser that can be used to visualise the
BAM outputs. Please check their <a href="http://www.broadinstitute.org/igv/">website</a>
for all the formats that IGV can display.</p>
<p>We will be using IGV later in the workshop for viewing a BAM file in the
genome browser. It requires the index of the BAM file to be in the same
folder as where the BAM file is. The index file should have the same
name as the BAM file and the suffix <code class="codehilite"><span class="na">.bai</span></code>. Finally, to create the index
of a BAM file you need to make sure that the file is sorted according to
chromosomal coordinates.</p>
<p>Sort alignments according to chromosomal position and store the result
in the file with the prefix <code class="codehilite">Blood.sorted</code>:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools sort SM_Blood.bam SM_Blood.sorted
</pre></div>
</td></tr></table>

<p><br>
Index the sorted file.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>samtools index SM_Blood.sorted.bam
</pre></div>
</td></tr></table>

<p>The indexing will create a file called <code class="codehilite">Blood.sorted.bam.bai</code>. Note that
you don’t have to specify the name of the index file when running
<code class="codehilite">samtools index</code>, it simply appends a <code class="codehilite"><span class="na">.bai</span></code> suffix to the input BAM
file.<br />
<br></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>How can you quickly find out whether a BAM file is already coordinate
sorted or not?</p>
<div class="admonition success"><details><summary><strong>Answer</strong></summary><p>Use samtools view command to look at the SAM header.</p></details></div>
</div>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../btp-module-ngs-bio/biol/" title="NGS Biological Insight" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                NGS Biological Insight
              </span>
            </div>
          </a>
        
        
          <a href="../../cancer-module-snv/snv/" title="Single Nucleotide Variant Calling and Annotation" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Single Nucleotide Variant Calling and Annotation
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-e3caa82af6.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>